{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "566fc2d0-d528-4b97-b05c-463ad4b0c816",
   "metadata": {},
   "outputs": [],
   "source": [
    "import numpy as np\n",
    "import pandas as pd\n",
    "import glob\n",
    "import os\n",
    "import sys\n",
    "from pandas import ExcelWriter"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "c85b08d9-e1d9-4771-8389-96b4c28f182e",
   "metadata": {},
   "outputs": [],
   "source": [
    "\n",
    "nv1=pd.read_csv('conjFDR_0.05_GGE_vs_COG_lead.csv',sep='\\t')\n",
    "\n",
    "nv1.head()\n",
    "\n",
    "gw1=pd.read_csv('gwascatalog.txt',sep='\\t')\n",
    "gw1.head()\n",
    "\n",
    "gw1.columns\n",
    "\n",
    "snp1=pd.read_csv('conjFDR_0.05_GGE_vs_COG_snps.csv',sep='\\t')\n",
    "snp1.head()\n",
    "\n",
    "db1=pd.read_csv('novelty_ep4.csv',sep='\\t')\n",
    "db1.head()\n",
    "\n",
    "nov='epilep'\n",
    "\n",
    "gw2=gw1[gw1['Trait'].str.contains(fr'{nov}(?!$)')]\n",
    "gw2.head()\n",
    "\n",
    "gw1.Trait.value_counts()\n",
    "\n",
    "#snp1['Novel_in_GWAScatalog'] = snp1.groupby('locusnum')['CAND_SNP'].apply(lambda s: s.isin(gw2['snp']))\n",
    "\n",
    "new_c=snp1.groupby('locusnum')['CAND_SNP'].apply(lambda s: s.isin(gw2['snp']))\n",
    "\n",
    "snp1['Novel_in_GWAScatalog']=new_c.reset_index(level=0, drop=True)\n",
    "\n",
    "snp1.head()\n",
    "\n",
    "nv1=nv1.merge(snp1.groupby('locusnum')['Novel_in_GWAScatalog'].any(), left_on='locusnum', right_index=True, how='left')\n",
    "\n",
    "nv1.Novel_in_GWAScatalog=nv1.Novel_in_GWAScatalog.map({True: 'No', False: 'Yes'})\n",
    "\n",
    "nv1['MinBP']=nv1['MinBP'].astype(int)\n",
    "nv1['MaxBP']=nv1['MaxBP'].astype(int)\n",
    "nv1.CHR=nv1.CHR.astype(int)\n",
    "db1=db1[db1.chromosome.astype('str').str.isnumeric()]\n",
    "db1.chromosome=db1.chromosome.astype(int)\n",
    "db1['min_bp']=db1['min_bp'].astype(int)\n",
    "db1['max_bp']=db1['max_bp'].astype(int)\n",
    "a = nv1['MinBP'].to_numpy()\n",
    "b = nv1['MaxBP'].to_numpy()\n",
    "c1= nv1.CHR.to_numpy()\n",
    "c2= db1.chromosome.to_numpy()[:,None]\n",
    "# Raise the columns in db1 by a dimension to enable numpy's broadcasting\n",
    "c = db1['min_bp'].to_numpy()[:, None]\n",
    "d = db1['max_bp'].to_numpy()[:, None]\n",
    "\n",
    "result = ((c1 == c2) & (a >= c) & (a <= d)) | ((c1 == c2) & (b >= c) & (b <= d)) | ((c1 == c2) & (a <= c) & (b >= d))\n",
    "\n",
    "nv1['Novel_in_Database'] = np.where(result.any(axis=0), 'No', 'Yes')\n",
    "\n",
    "trait1='GGE'\n",
    "trait2='COG'\n",
    "\n",
    "nv1[f'Novel_in_{trait1}']=np.where(((nv1.Novel_in_GWAScatalog==\"Yes\") & (nv1.Novel_in_Database=='Yes')),'Yes','No')\n",
    "\n",
    "nv1=nv1.drop(['Novel_in_GWAScatalog', 'Novel_in_Database'], axis = 1)\n",
    "\n",
    "nv1.to_csv(f'conjFDR_0.05_{trait1}_vs_{trait2}_novelty.csv',index=False,sep='\\t')\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 35,
   "id": "314224b9-a706-445a-aa06-ca1925b4753c",
   "metadata": {},
   "outputs": [],
   "source": [
    "nv1=pd.read_csv('conjFDR_0.05_GGE_vs_COG_lead.csv',sep='\\t')\n",
    "\n",
    "nv1.head()\n",
    "\n",
    "gw1=pd.read_csv('gwascatalog.txt',sep='\\t')\n",
    "gw1.head()\n",
    "\n",
    "gw1.columns\n",
    "\n",
    "snp1=pd.read_csv('conjFDR_0.05_GGE_vs_COG_snps.csv',sep='\\t')\n",
    "snp1.head()\n",
    "\n",
    "db1=pd.read_csv('novelty_checking_database_cog.csv',sep='\\t')\n",
    "db1.head()\n",
    "\n",
    "nov='cognit'\n",
    "\n",
    "gw2=gw1[gw1['Trait'].str.contains(fr'{nov}(?!$)')]\n",
    "gw2.head()\n",
    "\n",
    "gw1.Trait.value_counts()\n",
    "\n",
    "#snp1['Novel_in_GWAScatalog'] = snp1.groupby('locusnum')['CAND_SNP'].apply(lambda s: s.isin(gw2['snp']))\n",
    "\n",
    "new_c=snp1.groupby('locusnum')['CAND_SNP'].apply(lambda s: s.isin(gw2['snp']))\n",
    "\n",
    "snp1['Novel_in_GWAScatalog']=new_c.reset_index(level=0, drop=True)\n",
    "\n",
    "snp1.head()\n",
    "\n",
    "nv1=nv1.merge(snp1.groupby('locusnum')['Novel_in_GWAScatalog'].any(), left_on='locusnum', right_index=True, how='left')\n",
    "\n",
    "nv1.Novel_in_GWAScatalog=nv1.Novel_in_GWAScatalog.map({True: 'No', False: 'Yes'})\n",
    "\n",
    "nv1['MinBP']=nv1['MinBP'].astype(int)\n",
    "nv1['MaxBP']=nv1['MaxBP'].astype(int)\n",
    "nv1.CHR=nv1.CHR.astype(int)\n",
    "db1=db1[db1.chromosome.astype('str').str.isnumeric()]\n",
    "db1.chromosome=db1.chromosome.astype(int)\n",
    "db1['min_bp']=db1['min_bp'].astype(int)\n",
    "db1['max_bp']=db1['max_bp'].astype(int)\n",
    "a = nv1['MinBP'].to_numpy()\n",
    "b = nv1['MaxBP'].to_numpy()\n",
    "c1= nv1.CHR.to_numpy()\n",
    "c2= db1.chromosome.to_numpy()[:,None]\n",
    "# Raise the columns in db1 by a dimension to enable numpy's broadcasting\n",
    "c = db1['min_bp'].to_numpy()[:, None]\n",
    "d = db1['max_bp'].to_numpy()[:, None]\n",
    "\n",
    "result = ((c1 == c2) & (a >= c) & (a <= d)) | ((c1 == c2) & (b >= c) & (b <= d)) | ((c1 == c2) & (a <= c) & (b >= d))\n",
    "\n",
    "nv1['Novel_in_Database'] = np.where(result.any(axis=0), 'No', 'Yes')\n",
    "\n",
    "trait1='COG'\n",
    "trait2='GGE'\n",
    "\n",
    "nv1[f'Novel_in_{trait1}']=np.where(((nv1.Novel_in_GWAScatalog==\"Yes\") & (nv1.Novel_in_Database=='Yes')),'Yes','No')\n",
    "\n",
    "nv1=nv1.drop(['Novel_in_GWAScatalog', 'Novel_in_Database'], axis = 1)\n",
    "\n",
    "nv1.to_csv(f'conjFDR_0.05_{trait1}_vs_{trait2}_novelty.csv',index=False,sep='\\t')\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "34815479-5396-4d37-92a8-ef4a39cbe00e",
   "metadata": {},
   "outputs": [],
   "source": [
    "def novelty_check(lead,gwas,snp,db,nov,trait1,trait2):\n",
    "    nv1=pd.read_csv(lead,sep='\\t')\n",
    "    nv1.head()\n",
    "    gw1=pd.read_csv(gwas,sep='\\t')\n",
    "    gw1.head()\n",
    "\n",
    "    gw1.columns\n",
    "\n",
    "    snp1=pd.read_csv(snp,sep='\\t')\n",
    "    snp1.head()\n",
    "\n",
    "    db1=pd.read_csv(db,sep='\\t')\n",
    "    db1.head()\n",
    "\n",
    "    #nov='epilep'\n",
    "\n",
    "    gw2=gw1[gw1['Trait'].str.contains(fr'{nov}(?!$)')]\n",
    "    gw2.head()\n",
    "\n",
    "    gw1.Trait.value_counts()\n",
    "\n",
    "    #snp1['Novel_in_GWAScatalog'] = snp1.groupby('locusnum')['CAND_SNP'].apply(lambda s: s.isin(gw2['snp']))\n",
    "\n",
    "    new_col=snp1.groupby('locusnum')['CAND_SNP'].apply(lambda s: s.isin(gw2['snp']))\n",
    "\n",
    "    snp1['Novel_in_GWAScatalog']=new_col.reset_index(level=0, drop=True)\n",
    "\n",
    "    snp1.head()\n",
    "\n",
    "    nv1=nv1.merge(snp1.groupby('locusnum')['Novel_in_GWAScatalog'].any(), left_on='locusnum', right_index=True, how='left')\n",
    "\n",
    "    nv1.Novel_in_GWAScatalog=nv1.Novel_in_GWAScatalog.map({True: 'No', False: 'Yes'})\n",
    "\n",
    "    nv1['MinBP']=nv1['MinBP'].astype(int)\n",
    "    nv1['MaxBP']=nv1['MaxBP'].astype(int)\n",
    "    nv1.CHR=nv1.CHR.astype(int)\n",
    "    db1=db1[db1.chromosome.astype('str').str.isnumeric()]\n",
    "    db1.chromosome=db1.chromosome.astype(int)\n",
    "    db1['min_bp']=db1['min_bp'].astype(int)\n",
    "    db1['max_bp']=db1['max_bp'].astype(int)\n",
    "    a = nv1['MinBP'].to_numpy()\n",
    "    b = nv1['MaxBP'].to_numpy()\n",
    "    c1= nv1.CHR.to_numpy()\n",
    "    c2= db1.chromosome.to_numpy()[:,None]\n",
    "    # Raise the columns in db1 by a dimension to enable numpy's broadcasting\n",
    "    c = db1['min_bp'].to_numpy()[:, None]\n",
    "    d = db1['max_bp'].to_numpy()[:, None]\n",
    "\n",
    "    result = ((c1 == c2) & (a >= c) & (a <= d)) | ((c1 == c2) & (b >= c) & (b <= d)) | ((c1 == c2) & (a <= c) & (b >= d))\n",
    "\n",
    "    nv1['Novel_in_Database'] = np.where(result.any(axis=0), 'No', 'Yes')\n",
    "\n",
    "    #trait1='GGE'\n",
    "    #trait2='COG'\n",
    "\n",
    "    nv1[f'Novel_in_{trait1}']=np.where(((nv1.Novel_in_GWAScatalog==\"Yes\") & (nv1.Novel_in_Database=='Yes')),'Yes','No')\n",
    "\n",
    "    nv1=nv1.drop(['Novel_in_GWAScatalog', 'Novel_in_Database'], axis = 1)\n",
    "\n",
    "    nv1.to_csv(f'conjFDR_0.05_{trait1}_vs_{trait2}_novelty.csv',index=False,sep='\\t')\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "1c7f1e5d-f44e-4c00-bd16-0d923ea5c861",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'/Users/zillurrahman/Desktop/zillur/work/coauthorships/novelty_for_naz'"
      ]
     },
     "execution_count": 3,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "os.getcwd()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "id": "98a326f7-6568-4758-bb27-47e7d7cee926",
   "metadata": {},
   "outputs": [],
   "source": [
    "novelty_check('zillur/all/conjFDR_0.05_ALL_vs_COG_lead.csv','zillur/all/gwascatalog.txt','zillur/all/conjFDR_0.05_ALL_vs_COG_snps.csv',\n",
    "             'zillur/all/novelty_ep4.csv','epilep','All','COG')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "id": "df993a5e-02ad-42a0-9e92-8902af34b8f6",
   "metadata": {},
   "outputs": [],
   "source": [
    "novelty_check('zillur/all/conjFDR_0.05_ALL_vs_COG_lead.csv','zillur/all/gwascatalog.txt','zillur/all/conjFDR_0.05_ALL_vs_COG_snps.csv',\n",
    "             'zillur/all/novelty_checking_database_cog.csv','cognit','COG','All')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "id": "acdc7401-4054-4c98-b9f4-d2b39fb45782",
   "metadata": {},
   "outputs": [],
   "source": [
    "novelty_check('zillur/cae/conjFDR_0.05_CAE_vs_COG_lead.csv','zillur/cae/gwascatalog.txt','zillur/cae/conjFDR_0.05_CAE_vs_COG_snps.csv',\n",
    "             'zillur/cae/novelty_ep4.csv','epilep','CAE','COG')\n",
    "novelty_check('zillur/cae/conjFDR_0.05_CAE_vs_COG_lead.csv','zillur/cae/gwascatalog.txt','zillur/cae/conjFDR_0.05_CAE_vs_COG_snps.csv',\n",
    "             'zillur/cae/novelty_checking_database_cog.csv','cognit','COG','CAE')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "id": "2e9f5390-79b5-410a-9d82-4d436fc06f8d",
   "metadata": {},
   "outputs": [],
   "source": [
    "novelty_check('zillur/jme/conjFDR_0.05_JME_vs_COG_lead.csv','zillur/jme/gwascatalog.txt','zillur/jme/conjFDR_0.05_JME_vs_COG_snps.csv',\n",
    "             'zillur/jme/novelty_ep4.csv','epilep','JME','COG')\n",
    "novelty_check('zillur/jme/conjFDR_0.05_JME_vs_COG_lead.csv','zillur/jme/gwascatalog.txt','zillur/jme/conjFDR_0.05_JME_vs_COG_snps.csv',\n",
    "             'zillur/jme/novelty_checking_database_cog.csv','cognit','COG','JME')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "f1785c18-c502-4ada-be18-9e700c2167b0",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Data_set</th>\n",
       "      <th>Trait</th>\n",
       "      <th>Description</th>\n",
       "      <th>N</th>\n",
       "      <th>Ncase</th>\n",
       "      <th>Ncontrol</th>\n",
       "      <th>Covariates</th>\n",
       "      <th>Population</th>\n",
       "      <th>GENE</th>\n",
       "      <th>CHR</th>\n",
       "      <th>START</th>\n",
       "      <th>STOP</th>\n",
       "      <th>NSNPS</th>\n",
       "      <th>NPARAM</th>\n",
       "      <th>N.1</th>\n",
       "      <th>ZSTAT</th>\n",
       "      <th>P</th>\n",
       "      <th>P_adjusted</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>BCX</td>\n",
       "      <td>WBC</td>\n",
       "      <td>Publicely Available</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>KCNQ1</td>\n",
       "      <td>11.0</td>\n",
       "      <td>2466221.0</td>\n",
       "      <td>2870340.0</td>\n",
       "      <td>1490.0</td>\n",
       "      <td>245.0</td>\n",
       "      <td>518638.0</td>\n",
       "      <td>4.21900</td>\n",
       "      <td>1.230000e-01</td>\n",
       "      <td>3.005660e-04</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>CTG</td>\n",
       "      <td>INTELLIGENCE</td>\n",
       "      <td>Publicely Available</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>KCNQ1</td>\n",
       "      <td>11.0</td>\n",
       "      <td>2466221.0</td>\n",
       "      <td>2870340.0</td>\n",
       "      <td>1196.0</td>\n",
       "      <td>187.0</td>\n",
       "      <td>57547.0</td>\n",
       "      <td>0.59066</td>\n",
       "      <td>2.773700e-01</td>\n",
       "      <td>9.707950e-01</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>CTG</td>\n",
       "      <td>INSOMNIA</td>\n",
       "      <td>Publicely Available</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>KCNQ1</td>\n",
       "      <td>11.0</td>\n",
       "      <td>2466221.0</td>\n",
       "      <td>2870340.0</td>\n",
       "      <td>944.0</td>\n",
       "      <td>156.0</td>\n",
       "      <td>382837.0</td>\n",
       "      <td>0.15140</td>\n",
       "      <td>4.398300e-01</td>\n",
       "      <td>9.989800e-01</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>DIAMANTE</td>\n",
       "      <td>T2DM</td>\n",
       "      <td>Publicely Available</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>KCNQ1</td>\n",
       "      <td>11.0</td>\n",
       "      <td>2466221.0</td>\n",
       "      <td>2870340.0</td>\n",
       "      <td>1144.0</td>\n",
       "      <td>181.0</td>\n",
       "      <td>293100.0</td>\n",
       "      <td>14.18000</td>\n",
       "      <td>6.080000e-42</td>\n",
       "      <td>2.980000e-38</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>GSCAN</td>\n",
       "      <td>SMOKE</td>\n",
       "      <td>Publicely Available</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>KCNQ1</td>\n",
       "      <td>11.0</td>\n",
       "      <td>2466221.0</td>\n",
       "      <td>2870340.0</td>\n",
       "      <td>1388.0</td>\n",
       "      <td>234.0</td>\n",
       "      <td>227978.0</td>\n",
       "      <td>-0.35123</td>\n",
       "      <td>6.372900e-01</td>\n",
       "      <td>9.989800e-01</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   Data_set         Trait          Description   N Ncase Ncontrol Covariates  \\\n",
       "0       BCX           WBC  Publicely Available NaN   NaN      NaN        NaN   \n",
       "1       CTG  INTELLIGENCE  Publicely Available NaN   NaN      NaN        NaN   \n",
       "2       CTG      INSOMNIA  Publicely Available NaN   NaN      NaN        NaN   \n",
       "3  DIAMANTE         T2DM   Publicely Available NaN   NaN      NaN        NaN   \n",
       "4     GSCAN         SMOKE  Publicely Available NaN   NaN      NaN        NaN   \n",
       "\n",
       "  Population   GENE   CHR      START       STOP   NSNPS  NPARAM       N.1  \\\n",
       "0        NaN  KCNQ1  11.0  2466221.0  2870340.0  1490.0   245.0  518638.0   \n",
       "1        NaN  KCNQ1  11.0  2466221.0  2870340.0  1196.0   187.0   57547.0   \n",
       "2        NaN  KCNQ1  11.0  2466221.0  2870340.0   944.0   156.0  382837.0   \n",
       "3        NaN  KCNQ1  11.0  2466221.0  2870340.0  1144.0   181.0  293100.0   \n",
       "4        NaN  KCNQ1  11.0  2466221.0  2870340.0  1388.0   234.0  227978.0   \n",
       "\n",
       "      ZSTAT             P    P_adjusted  \n",
       "0   4.21900  1.230000e-01  3.005660e-04  \n",
       "1   0.59066  2.773700e-01  9.707950e-01  \n",
       "2   0.15140  4.398300e-01  9.989800e-01  \n",
       "3  14.18000  6.080000e-42  2.980000e-38  \n",
       "4  -0.35123  6.372900e-01  9.989800e-01  "
      ]
     },
     "execution_count": 3,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "hsk1 = pd.read_excel('../../husk_ment/prime/kcnq1_magma_report_v1_dec24.xlsx')\n",
    "hsk1.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "ccaa0c66",
   "metadata": {},
   "outputs": [],
   "source": [
    "from statsmodels.stats.multitest import multipletests"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "id": "aade9dad",
   "metadata": {},
   "outputs": [],
   "source": [
    "hsk1['P'] = hsk1['P'].astype(float)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "id": "c5a46ec2",
   "metadata": {},
   "outputs": [],
   "source": [
    "_, p_adjusted, _, _ = multipletests(hsk1['P'], method='fdr_bh')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "11208835",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "id": "671ec8aa",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Data_set</th>\n",
       "      <th>Trait</th>\n",
       "      <th>Description</th>\n",
       "      <th>N</th>\n",
       "      <th>Ncase</th>\n",
       "      <th>Ncontrol</th>\n",
       "      <th>Covariates</th>\n",
       "      <th>Population</th>\n",
       "      <th>GENE</th>\n",
       "      <th>CHR</th>\n",
       "      <th>START</th>\n",
       "      <th>STOP</th>\n",
       "      <th>NSNPS</th>\n",
       "      <th>NPARAM</th>\n",
       "      <th>N.1</th>\n",
       "      <th>ZSTAT</th>\n",
       "      <th>P</th>\n",
       "      <th>P_adjusted</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>BCX</td>\n",
       "      <td>WBC</td>\n",
       "      <td>Publicely Available</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>KCNQ1</td>\n",
       "      <td>11.0</td>\n",
       "      <td>2466221.0</td>\n",
       "      <td>2870340.0</td>\n",
       "      <td>1490.0</td>\n",
       "      <td>245.0</td>\n",
       "      <td>518638.0</td>\n",
       "      <td>4.21900</td>\n",
       "      <td>1.230000e-01</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>CTG</td>\n",
       "      <td>INTELLIGENCE</td>\n",
       "      <td>Publicely Available</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>KCNQ1</td>\n",
       "      <td>11.0</td>\n",
       "      <td>2466221.0</td>\n",
       "      <td>2870340.0</td>\n",
       "      <td>1196.0</td>\n",
       "      <td>187.0</td>\n",
       "      <td>57547.0</td>\n",
       "      <td>0.59066</td>\n",
       "      <td>2.773700e-01</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>CTG</td>\n",
       "      <td>INSOMNIA</td>\n",
       "      <td>Publicely Available</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>KCNQ1</td>\n",
       "      <td>11.0</td>\n",
       "      <td>2466221.0</td>\n",
       "      <td>2870340.0</td>\n",
       "      <td>944.0</td>\n",
       "      <td>156.0</td>\n",
       "      <td>382837.0</td>\n",
       "      <td>0.15140</td>\n",
       "      <td>4.398300e-01</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>DIAMANTE</td>\n",
       "      <td>T2DM</td>\n",
       "      <td>Publicely Available</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>KCNQ1</td>\n",
       "      <td>11.0</td>\n",
       "      <td>2466221.0</td>\n",
       "      <td>2870340.0</td>\n",
       "      <td>1144.0</td>\n",
       "      <td>181.0</td>\n",
       "      <td>293100.0</td>\n",
       "      <td>14.18000</td>\n",
       "      <td>6.080000e-42</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>GSCAN</td>\n",
       "      <td>SMOKE</td>\n",
       "      <td>Publicely Available</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>KCNQ1</td>\n",
       "      <td>11.0</td>\n",
       "      <td>2466221.0</td>\n",
       "      <td>2870340.0</td>\n",
       "      <td>1388.0</td>\n",
       "      <td>234.0</td>\n",
       "      <td>227978.0</td>\n",
       "      <td>-0.35123</td>\n",
       "      <td>6.372900e-01</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   Data_set         Trait          Description   N Ncase Ncontrol Covariates  \\\n",
       "0       BCX           WBC  Publicely Available NaN   NaN      NaN        NaN   \n",
       "1       CTG  INTELLIGENCE  Publicely Available NaN   NaN      NaN        NaN   \n",
       "2       CTG      INSOMNIA  Publicely Available NaN   NaN      NaN        NaN   \n",
       "3  DIAMANTE         T2DM   Publicely Available NaN   NaN      NaN        NaN   \n",
       "4     GSCAN         SMOKE  Publicely Available NaN   NaN      NaN        NaN   \n",
       "\n",
       "  Population   GENE   CHR      START       STOP   NSNPS  NPARAM       N.1  \\\n",
       "0        NaN  KCNQ1  11.0  2466221.0  2870340.0  1490.0   245.0  518638.0   \n",
       "1        NaN  KCNQ1  11.0  2466221.0  2870340.0  1196.0   187.0   57547.0   \n",
       "2        NaN  KCNQ1  11.0  2466221.0  2870340.0   944.0   156.0  382837.0   \n",
       "3        NaN  KCNQ1  11.0  2466221.0  2870340.0  1144.0   181.0  293100.0   \n",
       "4        NaN  KCNQ1  11.0  2466221.0  2870340.0  1388.0   234.0  227978.0   \n",
       "\n",
       "      ZSTAT             P  P_adjusted  \n",
       "0   4.21900  1.230000e-01         NaN  \n",
       "1   0.59066  2.773700e-01         NaN  \n",
       "2   0.15140  4.398300e-01         NaN  \n",
       "3  14.18000  6.080000e-42         NaN  \n",
       "4  -0.35123  6.372900e-01         NaN  "
      ]
     },
     "execution_count": 9,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "hsk1['P_adjusted'] = p_adjusted\n",
    "hsk1.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "id": "f6f13cd0",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(77, 18)"
      ]
     },
     "execution_count": 10,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "hsk1.shape"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "id": "530fcb52",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Index(['Data_set', 'Trait', 'Description', 'N', 'Ncase', 'Ncontrol',\n",
       "       'Covariates', 'Population', 'GENE', 'CHR', 'START', 'STOP', 'NSNPS',\n",
       "       'NPARAM', 'N.1', 'ZSTAT', 'P', 'P_adjusted'],\n",
       "      dtype='object')"
      ]
     },
     "execution_count": 12,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "hsk1.columns"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "id": "a8eedda1",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Data_set</th>\n",
       "      <th>Trait</th>\n",
       "      <th>Description</th>\n",
       "      <th>N</th>\n",
       "      <th>Ncase</th>\n",
       "      <th>Ncontrol</th>\n",
       "      <th>Covariates</th>\n",
       "      <th>Population</th>\n",
       "      <th>GENE</th>\n",
       "      <th>CHR</th>\n",
       "      <th>START</th>\n",
       "      <th>STOP</th>\n",
       "      <th>NSNPS</th>\n",
       "      <th>NPARAM</th>\n",
       "      <th>N.1</th>\n",
       "      <th>ZSTAT</th>\n",
       "      <th>P</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>BCX</td>\n",
       "      <td>WBC</td>\n",
       "      <td>Publicely Available</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>KCNQ1</td>\n",
       "      <td>11.0</td>\n",
       "      <td>2466221.0</td>\n",
       "      <td>2870340.0</td>\n",
       "      <td>1490.0</td>\n",
       "      <td>245.0</td>\n",
       "      <td>518638.0</td>\n",
       "      <td>4.21900</td>\n",
       "      <td>1.230000e-01</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>CTG</td>\n",
       "      <td>INTELLIGENCE</td>\n",
       "      <td>Publicely Available</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>KCNQ1</td>\n",
       "      <td>11.0</td>\n",
       "      <td>2466221.0</td>\n",
       "      <td>2870340.0</td>\n",
       "      <td>1196.0</td>\n",
       "      <td>187.0</td>\n",
       "      <td>57547.0</td>\n",
       "      <td>0.59066</td>\n",
       "      <td>2.773700e-01</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>CTG</td>\n",
       "      <td>INSOMNIA</td>\n",
       "      <td>Publicely Available</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>KCNQ1</td>\n",
       "      <td>11.0</td>\n",
       "      <td>2466221.0</td>\n",
       "      <td>2870340.0</td>\n",
       "      <td>944.0</td>\n",
       "      <td>156.0</td>\n",
       "      <td>382837.0</td>\n",
       "      <td>0.15140</td>\n",
       "      <td>4.398300e-01</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>DIAMANTE</td>\n",
       "      <td>T2DM</td>\n",
       "      <td>Publicely Available</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>KCNQ1</td>\n",
       "      <td>11.0</td>\n",
       "      <td>2466221.0</td>\n",
       "      <td>2870340.0</td>\n",
       "      <td>1144.0</td>\n",
       "      <td>181.0</td>\n",
       "      <td>293100.0</td>\n",
       "      <td>14.18000</td>\n",
       "      <td>6.080000e-42</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>GSCAN</td>\n",
       "      <td>SMOKE</td>\n",
       "      <td>Publicely Available</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>KCNQ1</td>\n",
       "      <td>11.0</td>\n",
       "      <td>2466221.0</td>\n",
       "      <td>2870340.0</td>\n",
       "      <td>1388.0</td>\n",
       "      <td>234.0</td>\n",
       "      <td>227978.0</td>\n",
       "      <td>-0.35123</td>\n",
       "      <td>6.372900e-01</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   Data_set         Trait          Description   N Ncase Ncontrol Covariates  \\\n",
       "0       BCX           WBC  Publicely Available NaN   NaN      NaN        NaN   \n",
       "1       CTG  INTELLIGENCE  Publicely Available NaN   NaN      NaN        NaN   \n",
       "2       CTG      INSOMNIA  Publicely Available NaN   NaN      NaN        NaN   \n",
       "3  DIAMANTE         T2DM   Publicely Available NaN   NaN      NaN        NaN   \n",
       "4     GSCAN         SMOKE  Publicely Available NaN   NaN      NaN        NaN   \n",
       "\n",
       "  Population   GENE   CHR      START       STOP   NSNPS  NPARAM       N.1  \\\n",
       "0        NaN  KCNQ1  11.0  2466221.0  2870340.0  1490.0   245.0  518638.0   \n",
       "1        NaN  KCNQ1  11.0  2466221.0  2870340.0  1196.0   187.0   57547.0   \n",
       "2        NaN  KCNQ1  11.0  2466221.0  2870340.0   944.0   156.0  382837.0   \n",
       "3        NaN  KCNQ1  11.0  2466221.0  2870340.0  1144.0   181.0  293100.0   \n",
       "4        NaN  KCNQ1  11.0  2466221.0  2870340.0  1388.0   234.0  227978.0   \n",
       "\n",
       "      ZSTAT             P  \n",
       "0   4.21900  1.230000e-01  \n",
       "1   0.59066  2.773700e-01  \n",
       "2   0.15140  4.398300e-01  \n",
       "3  14.18000  6.080000e-42  \n",
       "4  -0.35123  6.372900e-01  "
      ]
     },
     "execution_count": 13,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "hsk2 = hsk1.drop('P_adjusted',axis=1)\n",
    "hsk2.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "id": "e5c8f3b8",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "<class 'pandas.core.frame.DataFrame'>\n",
      "RangeIndex: 77 entries, 0 to 76\n",
      "Data columns (total 17 columns):\n",
      " #   Column       Non-Null Count  Dtype  \n",
      "---  ------       --------------  -----  \n",
      " 0   Data_set     67 non-null     object \n",
      " 1   Trait        58 non-null     object \n",
      " 2   Description  58 non-null     object \n",
      " 3   N            36 non-null     float64\n",
      " 4   Ncase        36 non-null     object \n",
      " 5   Ncontrol     36 non-null     object \n",
      " 6   Covariates   36 non-null     object \n",
      " 7   Population   36 non-null     object \n",
      " 8   GENE         58 non-null     object \n",
      " 9   CHR          58 non-null     float64\n",
      " 10  START        58 non-null     float64\n",
      " 11  STOP         58 non-null     float64\n",
      " 12  NSNPS        58 non-null     float64\n",
      " 13  NPARAM       58 non-null     float64\n",
      " 14  N.1          58 non-null     float64\n",
      " 15  ZSTAT        58 non-null     float64\n",
      " 16  P            58 non-null     float64\n",
      "dtypes: float64(9), object(8)\n",
      "memory usage: 10.4+ KB\n"
     ]
    }
   ],
   "source": [
    "hsk2.info()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "id": "7a87182b",
   "metadata": {},
   "outputs": [],
   "source": [
    "_, p_adjusted, _, _ = multipletests(hsk2['P'], method='fdr_bh')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "id": "955507f4",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "array([nan, nan, nan, nan, nan, nan, nan, nan, nan, nan, nan, nan, nan,\n",
       "       nan, nan, nan, nan, nan, nan, nan, nan, nan, nan, nan, nan, nan,\n",
       "       nan, nan, nan, nan, nan, nan, nan, nan, nan, nan, nan, nan, nan,\n",
       "       nan, nan, nan, nan, nan, nan, nan, nan, nan, nan, nan, nan, nan,\n",
       "       nan, nan, nan, nan, nan, nan, nan, nan, nan, nan, nan, nan, nan,\n",
       "       nan, nan, nan, nan, nan, nan, nan, nan, nan, nan, nan, nan])"
      ]
     },
     "execution_count": 16,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "p_adjusted"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "id": "6b6cb32f",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "238200"
      ]
     },
     "execution_count": 19,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "87600*2 + 63000"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "id": "0ab058ae",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "234724.1"
      ]
     },
     "execution_count": 20,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "21242*11.05"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "id": "82ed2835",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Collecting PyMuPDF\n",
      "  Downloading pymupdf-1.25.1-cp39-abi3-macosx_11_0_arm64.whl.metadata (3.4 kB)\n",
      "Downloading pymupdf-1.25.1-cp39-abi3-macosx_11_0_arm64.whl (18.6 MB)\n",
      "\u001b[2K   \u001b[90m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━\u001b[0m \u001b[32m18.6/18.6 MB\u001b[0m \u001b[31m6.5 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0ma \u001b[36m0:00:01\u001b[0m\n",
      "\u001b[?25hInstalling collected packages: PyMuPDF\n",
      "Successfully installed PyMuPDF-1.25.1\n"
     ]
    }
   ],
   "source": [
    "!pip install PyMuPDF"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "id": "eb4647b8",
   "metadata": {},
   "outputs": [],
   "source": [
    "import fitz"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 40,
   "id": "4bc7f79b",
   "metadata": {},
   "outputs": [],
   "source": [
    "def pdf_to_png(pdf_path, output_folder,name, zoom=2.0, rotation=0):\n",
    "    # Open the PDF file\n",
    "    pdf_document = fitz.open(pdf_path)\n",
    "    \n",
    "    # Iterate over each page\n",
    "    for page_number in range(len(pdf_document)):\n",
    "        # Select the page\n",
    "        page = pdf_document.load_page(page_number)\n",
    "        \n",
    "        # Create the transform for scaling\n",
    "        matrix = fitz.Matrix(zoom, zoom)\n",
    "        \n",
    "        # Render the page to a pixmap\n",
    "        pixmap = page.get_pixmap(matrix=matrix)\n",
    "        \n",
    "        # Define the output path for the PNG file\n",
    "        output_path = f\"{output_folder}/{name}_photo_{page_number + 1}.jpeg\"\n",
    "        \n",
    "        # Save the image as a PNG file\n",
    "        pixmap.save(output_path)\n",
    "        \n",
    "    print(\"PDF has been converted to PNG and saved in the specified folder.\")\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 37,
   "id": "23a9186a",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "PDF has been converted to PNG and saved in the specified folder.\n"
     ]
    }
   ],
   "source": [
    "pdf_to_png('/Users/zillurrahman/Documents/personal/zillur-new-passport-2022-2032.pdf',\n",
    "           '/Users/zillurrahman/Documents/personal','zillur')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 38,
   "id": "0363ca51",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "PDF has been converted to PNG and saved in the specified folder.\n"
     ]
    }
   ],
   "source": [
    "pdf_to_png('/Users/zillurrahman/Documents/personal/all_documents_from_oslo/nasrin_akter/nupa_passport_visa.pdf',\n",
    "           '/Users/zillurrahman/Documents/personal','nupa')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 39,
   "id": "d85a9651",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "PDF has been converted to PNG and saved in the specified folder.\n"
     ]
    }
   ],
   "source": [
    "pdf_to_png('/Users/zillurrahman/Documents/renewal_residence/sannidho/sannidho_passport.pdf',\n",
    "           '/Users/zillurrahman/Documents/personal','sannidho')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 41,
   "id": "59c353a1",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "PDF has been converted to PNG and saved in the specified folder.\n"
     ]
    }
   ],
   "source": [
    "pdf_to_png('/Users/zillurrahman/Documents/personal/zillur_photo_pp.pdf',\n",
    "           '/Users/zillurrahman/Documents/personal','zillur')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 42,
   "id": "448217f5",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "PDF has been converted to PNG and saved in the specified folder.\n"
     ]
    }
   ],
   "source": [
    "pdf_to_png('/Users/zillurrahman/Documents/personal/nupa_photo_pp.pdf',\n",
    "           '/Users/zillurrahman/Documents/personal','nupa')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "id": "28ec71de",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Collecting gdown\n",
      "  Downloading gdown-5.2.0-py3-none-any.whl.metadata (5.8 kB)\n",
      "Requirement already satisfied: beautifulsoup4 in /Users/zillurrahman/miniconda3/lib/python3.10/site-packages (from gdown) (4.12.2)\n",
      "Requirement already satisfied: filelock in /Users/zillurrahman/miniconda3/lib/python3.10/site-packages (from gdown) (3.16.1)\n",
      "Requirement already satisfied: requests[socks] in /Users/zillurrahman/miniconda3/lib/python3.10/site-packages (from gdown) (2.32.3)\n",
      "Requirement already satisfied: tqdm in /Users/zillurrahman/miniconda3/lib/python3.10/site-packages (from gdown) (4.65.0)\n",
      "Requirement already satisfied: soupsieve>1.2 in /Users/zillurrahman/miniconda3/lib/python3.10/site-packages (from beautifulsoup4->gdown) (2.4.1)\n",
      "Requirement already satisfied: charset-normalizer<4,>=2 in /Users/zillurrahman/miniconda3/lib/python3.10/site-packages (from requests[socks]->gdown) (2.0.4)\n",
      "Requirement already satisfied: idna<4,>=2.5 in /Users/zillurrahman/miniconda3/lib/python3.10/site-packages (from requests[socks]->gdown) (3.4)\n",
      "Requirement already satisfied: urllib3<3,>=1.21.1 in /Users/zillurrahman/miniconda3/lib/python3.10/site-packages (from requests[socks]->gdown) (2.2.2)\n",
      "Requirement already satisfied: certifi>=2017.4.17 in /Users/zillurrahman/miniconda3/lib/python3.10/site-packages (from requests[socks]->gdown) (2022.12.7)\n",
      "Requirement already satisfied: PySocks!=1.5.7,>=1.5.6 in /Users/zillurrahman/miniconda3/lib/python3.10/site-packages (from requests[socks]->gdown) (1.7.1)\n",
      "Downloading gdown-5.2.0-py3-none-any.whl (18 kB)\n",
      "Installing collected packages: gdown\n",
      "Successfully installed gdown-5.2.0\n"
     ]
    }
   ],
   "source": [
    "!pip install gdown"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "id": "df965b79",
   "metadata": {},
   "outputs": [],
   "source": [
    "import gdown"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "id": "7ee894a8",
   "metadata": {},
   "outputs": [],
   "source": [
    "from io import BytesIO"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "id": "77f70030",
   "metadata": {},
   "outputs": [],
   "source": [
    "link1 = 'https://docs.google.com/spreadsheets/d/1PsAcFRlG0x_JZOknWLQ_GQiCyR_p5ZDiaR8TqR2NwoQ/edit?usp=sharing'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "id": "d4a2546f",
   "metadata": {},
   "outputs": [],
   "source": [
    "link2 = f'https://drive.google.com/uc?export=download&id=1PsAcFRlG0x_JZOknWLQ_GQiCyR_p5ZDiaR8TqR2NwoQ'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "id": "5eeba6ae",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/zillurrahman/miniconda3/lib/python3.10/site-packages/gdown/download.py:33: MarkupResemblesLocatorWarning: The input looks more like a filename than markup. You may want to open this file and pass the filehandle into Beautiful Soup.\n",
      "  soup = bs4.BeautifulSoup(line, features=\"html.parser\")\n"
     ]
    },
    {
     "ename": "FileURLRetrievalError",
     "evalue": "Failed to retrieve file url:\n\n\tCannot retrieve the public link of the file. You may need to change\n\tthe permission to 'Anyone with the link', or have had many accesses.\n\tCheck FAQ in https://github.com/wkentaro/gdown?tab=readme-ov-file#faq.\n\nYou may still be able to access the file from the browser:\n\n\thttps://drive.google.com/uc?export=download&id=1PsAcFRlG0x_JZOknWLQ_GQiCyR_p5ZDiaR8TqR2NwoQ\n\nbut Gdown can't. Please check connections and permissions.",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mFileURLRetrievalError\u001b[0m                     Traceback (most recent call last)",
      "File \u001b[0;32m~/miniconda3/lib/python3.10/site-packages/gdown/download.py:267\u001b[0m, in \u001b[0;36mdownload\u001b[0;34m(url, output, quiet, proxy, speed, use_cookies, verify, id, fuzzy, resume, format, user_agent, log_messages)\u001b[0m\n\u001b[1;32m    266\u001b[0m \u001b[38;5;28;01mtry\u001b[39;00m:\n\u001b[0;32m--> 267\u001b[0m     url \u001b[38;5;241m=\u001b[39m \u001b[43mget_url_from_gdrive_confirmation\u001b[49m\u001b[43m(\u001b[49m\u001b[43mres\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mtext\u001b[49m\u001b[43m)\u001b[49m\n\u001b[1;32m    268\u001b[0m \u001b[38;5;28;01mexcept\u001b[39;00m FileURLRetrievalError \u001b[38;5;28;01mas\u001b[39;00m e:\n",
      "File \u001b[0;32m~/miniconda3/lib/python3.10/site-packages/gdown/download.py:55\u001b[0m, in \u001b[0;36mget_url_from_gdrive_confirmation\u001b[0;34m(contents)\u001b[0m\n\u001b[1;32m     54\u001b[0m \u001b[38;5;28;01mif\u001b[39;00m \u001b[38;5;129;01mnot\u001b[39;00m url:\n\u001b[0;32m---> 55\u001b[0m     \u001b[38;5;28;01mraise\u001b[39;00m FileURLRetrievalError(\n\u001b[1;32m     56\u001b[0m         \u001b[38;5;124m\"\u001b[39m\u001b[38;5;124mCannot retrieve the public link of the file. \u001b[39m\u001b[38;5;124m\"\u001b[39m\n\u001b[1;32m     57\u001b[0m         \u001b[38;5;124m\"\u001b[39m\u001b[38;5;124mYou may need to change the permission to \u001b[39m\u001b[38;5;124m\"\u001b[39m\n\u001b[1;32m     58\u001b[0m         \u001b[38;5;124m\"\u001b[39m\u001b[38;5;124m'\u001b[39m\u001b[38;5;124mAnyone with the link\u001b[39m\u001b[38;5;124m'\u001b[39m\u001b[38;5;124m, or have had many accesses. \u001b[39m\u001b[38;5;124m\"\u001b[39m\n\u001b[1;32m     59\u001b[0m         \u001b[38;5;124m\"\u001b[39m\u001b[38;5;124mCheck FAQ in https://github.com/wkentaro/gdown?tab=readme-ov-file#faq.\u001b[39m\u001b[38;5;124m\"\u001b[39m,\n\u001b[1;32m     60\u001b[0m     )\n\u001b[1;32m     61\u001b[0m \u001b[38;5;28;01mreturn\u001b[39;00m url\n",
      "\u001b[0;31mFileURLRetrievalError\u001b[0m: Cannot retrieve the public link of the file. You may need to change the permission to 'Anyone with the link', or have had many accesses. Check FAQ in https://github.com/wkentaro/gdown?tab=readme-ov-file#faq.",
      "\nDuring handling of the above exception, another exception occurred:\n",
      "\u001b[0;31mFileURLRetrievalError\u001b[0m                     Traceback (most recent call last)",
      "Cell \u001b[0;32mIn[14], line 1\u001b[0m\n\u001b[0;32m----> 1\u001b[0m db1 \u001b[38;5;241m=\u001b[39m pd\u001b[38;5;241m.\u001b[39mread_excel(BytesIO(\u001b[43mgdown\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mdownload\u001b[49m\u001b[43m(\u001b[49m\u001b[43mlink2\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43mquiet\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[38;5;28;43;01mTrue\u001b[39;49;00m\u001b[43m)\u001b[49m),engine\u001b[38;5;241m=\u001b[39m\u001b[38;5;124m'\u001b[39m\u001b[38;5;124mopenpyxl\u001b[39m\u001b[38;5;124m'\u001b[39m)\n",
      "File \u001b[0;32m~/miniconda3/lib/python3.10/site-packages/gdown/download.py:278\u001b[0m, in \u001b[0;36mdownload\u001b[0;34m(url, output, quiet, proxy, speed, use_cookies, verify, id, fuzzy, resume, format, user_agent, log_messages)\u001b[0m\n\u001b[1;32m    268\u001b[0m     \u001b[38;5;28;01mexcept\u001b[39;00m FileURLRetrievalError \u001b[38;5;28;01mas\u001b[39;00m e:\n\u001b[1;32m    269\u001b[0m         message \u001b[38;5;241m=\u001b[39m (\n\u001b[1;32m    270\u001b[0m             \u001b[38;5;124m\"\u001b[39m\u001b[38;5;124mFailed to retrieve file url:\u001b[39m\u001b[38;5;130;01m\\n\u001b[39;00m\u001b[38;5;130;01m\\n\u001b[39;00m\u001b[38;5;132;01m{}\u001b[39;00m\u001b[38;5;130;01m\\n\u001b[39;00m\u001b[38;5;130;01m\\n\u001b[39;00m\u001b[38;5;124m\"\u001b[39m\n\u001b[1;32m    271\u001b[0m             \u001b[38;5;124m\"\u001b[39m\u001b[38;5;124mYou may still be able to access the file from the browser:\u001b[39m\u001b[38;5;124m\"\u001b[39m\n\u001b[0;32m   (...)\u001b[0m\n\u001b[1;32m    276\u001b[0m             url_origin,\n\u001b[1;32m    277\u001b[0m         )\n\u001b[0;32m--> 278\u001b[0m         \u001b[38;5;28;01mraise\u001b[39;00m FileURLRetrievalError(message)\n\u001b[1;32m    280\u001b[0m filename_from_url \u001b[38;5;241m=\u001b[39m \u001b[38;5;28;01mNone\u001b[39;00m\n\u001b[1;32m    281\u001b[0m last_modified_time \u001b[38;5;241m=\u001b[39m \u001b[38;5;28;01mNone\u001b[39;00m\n",
      "\u001b[0;31mFileURLRetrievalError\u001b[0m: Failed to retrieve file url:\n\n\tCannot retrieve the public link of the file. You may need to change\n\tthe permission to 'Anyone with the link', or have had many accesses.\n\tCheck FAQ in https://github.com/wkentaro/gdown?tab=readme-ov-file#faq.\n\nYou may still be able to access the file from the browser:\n\n\thttps://drive.google.com/uc?export=download&id=1PsAcFRlG0x_JZOknWLQ_GQiCyR_p5ZDiaR8TqR2NwoQ\n\nbut Gdown can't. Please check connections and permissions."
     ]
    }
   ],
   "source": [
    "db1 = pd.read_excel(BytesIO(gdown.download(link2, quiet=True)),engine='openpyxl')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "id": "0cd1be6d",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'/Users/zillurrahman/Desktop/zillur/work/body-brain/borge'"
      ]
     },
     "execution_count": 15,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "os.getcwd()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 26,
   "id": "238b3ef0",
   "metadata": {},
   "outputs": [],
   "source": [
    "db1 = pd.read_excel('novelty_checking_database.xlsx',engine='openpyxl',sheet_name=None)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 27,
   "id": "02cb1b05",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Alcohol use disorder</th>\n",
       "      <th>Unnamed: 1</th>\n",
       "      <th>Unnamed: 2</th>\n",
       "      <th>Unnamed: 3</th>\n",
       "      <th>Unnamed: 4</th>\n",
       "      <th>Unnamed: 5</th>\n",
       "      <th>Unnamed: 6</th>\n",
       "      <th>Unnamed: 7</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>Date of last update</td>\n",
       "      <td>04.05.2023</td>\n",
       "      <td>KEY</td>\n",
       "      <td>study = abbreviated study name as defined at t...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>Description of search completed</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>method = GWAS/condFDR/conjFDR/MTAG/STAR/other</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>Last update by</td>\n",
       "      <td>Børge</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>Search terms for literature search</td>\n",
       "      <td>Alcohol use disorder + genetic loci</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>List of studies included (author year journal ...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                Alcohol use disorder  \\\n",
       "0                                Date of last update   \n",
       "1                    Description of search completed   \n",
       "2                                     Last update by   \n",
       "3                 Search terms for literature search   \n",
       "4  List of studies included (author year journal ...   \n",
       "\n",
       "                            Unnamed: 1 Unnamed: 2  \\\n",
       "0                           04.05.2023        KEY   \n",
       "1                                  NaN        NaN   \n",
       "2                                Børge        NaN   \n",
       "3  Alcohol use disorder + genetic loci        NaN   \n",
       "4                                  NaN        NaN   \n",
       "\n",
       "                                          Unnamed: 3 Unnamed: 4 Unnamed: 5  \\\n",
       "0  study = abbreviated study name as defined at t...        NaN        NaN   \n",
       "1      method = GWAS/condFDR/conjFDR/MTAG/STAR/other        NaN        NaN   \n",
       "2                                                NaN        NaN        NaN   \n",
       "3                                                NaN        NaN        NaN   \n",
       "4                                                NaN        NaN        NaN   \n",
       "\n",
       "  Unnamed: 6 Unnamed: 7  \n",
       "0        NaN        NaN  \n",
       "1        NaN        NaN  \n",
       "2        NaN        NaN  \n",
       "3        NaN        NaN  \n",
       "4        NaN        NaN  "
      ]
     },
     "execution_count": 27,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "aud_db1 = db1.get('aud')\n",
    "aud_db1.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 31,
   "id": "a74123b3",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Alcohol use disorder</th>\n",
       "      <th>Unnamed: 1</th>\n",
       "      <th>Unnamed: 2</th>\n",
       "      <th>Unnamed: 3</th>\n",
       "      <th>Unnamed: 4</th>\n",
       "      <th>Unnamed: 5</th>\n",
       "      <th>Unnamed: 6</th>\n",
       "      <th>Unnamed: 7</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chromosome</td>\n",
       "      <td>min_bp</td>\n",
       "      <td>max_bp</td>\n",
       "      <td>lead_SNP</td>\n",
       "      <td>study</td>\n",
       "      <td>method</td>\n",
       "      <td>ethnicity</td>\n",
       "      <td>Pos</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>rs11210887</td>\n",
       "      <td>Xu2023</td>\n",
       "      <td>MTAG</td>\n",
       "      <td>EUR</td>\n",
       "      <td>44076019</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>rs2310819</td>\n",
       "      <td>Xu2023</td>\n",
       "      <td>MTAG</td>\n",
       "      <td>EUR</td>\n",
       "      <td>66440096</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>rs11210229</td>\n",
       "      <td>Xu2023</td>\n",
       "      <td>MTAG</td>\n",
       "      <td>EUR</td>\n",
       "      <td>73860028</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>rs1526480</td>\n",
       "      <td>Xu2023</td>\n",
       "      <td>MTAG</td>\n",
       "      <td>EUR</td>\n",
       "      <td>91209986</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "  Alcohol use disorder Unnamed: 1 Unnamed: 2  Unnamed: 3 Unnamed: 4  \\\n",
       "0           chromosome     min_bp     max_bp    lead_SNP      study   \n",
       "1                    1        NaN        NaN  rs11210887     Xu2023   \n",
       "2                    1        NaN        NaN   rs2310819     Xu2023   \n",
       "3                    1        NaN        NaN  rs11210229     Xu2023   \n",
       "4                    1        NaN        NaN   rs1526480     Xu2023   \n",
       "\n",
       "  Unnamed: 5 Unnamed: 6 Unnamed: 7  \n",
       "0     method  ethnicity        Pos  \n",
       "1       MTAG        EUR   44076019  \n",
       "2       MTAG        EUR   66440096  \n",
       "3       MTAG        EUR   73860028  \n",
       "4       MTAG        EUR   91209986  "
      ]
     },
     "execution_count": 31,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "aud_db2=aud_db1.iloc[9:].reset_index(drop=True)\n",
    "aud_db2.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 32,
   "id": "7121e552",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Alcohol use disorder    chromosome\n",
       "Unnamed: 1                  min_bp\n",
       "Unnamed: 2                  max_bp\n",
       "Unnamed: 3                lead_SNP\n",
       "Unnamed: 4                   study\n",
       "Unnamed: 5                  method\n",
       "Unnamed: 6               ethnicity\n",
       "Unnamed: 7                     Pos\n",
       "Name: 0, dtype: object"
      ]
     },
     "execution_count": 32,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "new_column_names1 = aud_db2.iloc[0]\n",
    "new_column_names1"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 33,
   "id": "c4ee4dbe",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Alcohol use disorder</th>\n",
       "      <th>Unnamed: 1</th>\n",
       "      <th>Unnamed: 2</th>\n",
       "      <th>Unnamed: 3</th>\n",
       "      <th>Unnamed: 4</th>\n",
       "      <th>Unnamed: 5</th>\n",
       "      <th>Unnamed: 6</th>\n",
       "      <th>Unnamed: 7</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>rs11210887</td>\n",
       "      <td>Xu2023</td>\n",
       "      <td>MTAG</td>\n",
       "      <td>EUR</td>\n",
       "      <td>44076019</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>rs2310819</td>\n",
       "      <td>Xu2023</td>\n",
       "      <td>MTAG</td>\n",
       "      <td>EUR</td>\n",
       "      <td>66440096</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>rs11210229</td>\n",
       "      <td>Xu2023</td>\n",
       "      <td>MTAG</td>\n",
       "      <td>EUR</td>\n",
       "      <td>73860028</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>rs1526480</td>\n",
       "      <td>Xu2023</td>\n",
       "      <td>MTAG</td>\n",
       "      <td>EUR</td>\n",
       "      <td>91209986</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>rs3897854</td>\n",
       "      <td>Xu2023</td>\n",
       "      <td>MTAG</td>\n",
       "      <td>EUR</td>\n",
       "      <td>97918306</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "  Alcohol use disorder Unnamed: 1 Unnamed: 2  Unnamed: 3 Unnamed: 4  \\\n",
       "0                    1        NaN        NaN  rs11210887     Xu2023   \n",
       "1                    1        NaN        NaN   rs2310819     Xu2023   \n",
       "2                    1        NaN        NaN  rs11210229     Xu2023   \n",
       "3                    1        NaN        NaN   rs1526480     Xu2023   \n",
       "4                    1        NaN        NaN   rs3897854     Xu2023   \n",
       "\n",
       "  Unnamed: 5 Unnamed: 6 Unnamed: 7  \n",
       "0       MTAG        EUR   44076019  \n",
       "1       MTAG        EUR   66440096  \n",
       "2       MTAG        EUR   73860028  \n",
       "3       MTAG        EUR   91209986  \n",
       "4       MTAG        EUR   97918306  "
      ]
     },
     "execution_count": 33,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "aud_db3 = aud_db2[1:].reset_index(drop=True)\n",
    "aud_db3.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 34,
   "id": "bdfc7bab",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>chromosome</th>\n",
       "      <th>min_bp</th>\n",
       "      <th>max_bp</th>\n",
       "      <th>lead_SNP</th>\n",
       "      <th>study</th>\n",
       "      <th>method</th>\n",
       "      <th>ethnicity</th>\n",
       "      <th>Pos</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>rs11210887</td>\n",
       "      <td>Xu2023</td>\n",
       "      <td>MTAG</td>\n",
       "      <td>EUR</td>\n",
       "      <td>44076019</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>rs2310819</td>\n",
       "      <td>Xu2023</td>\n",
       "      <td>MTAG</td>\n",
       "      <td>EUR</td>\n",
       "      <td>66440096</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>rs11210229</td>\n",
       "      <td>Xu2023</td>\n",
       "      <td>MTAG</td>\n",
       "      <td>EUR</td>\n",
       "      <td>73860028</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>rs1526480</td>\n",
       "      <td>Xu2023</td>\n",
       "      <td>MTAG</td>\n",
       "      <td>EUR</td>\n",
       "      <td>91209986</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>rs3897854</td>\n",
       "      <td>Xu2023</td>\n",
       "      <td>MTAG</td>\n",
       "      <td>EUR</td>\n",
       "      <td>97918306</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "0 chromosome min_bp max_bp    lead_SNP   study method ethnicity       Pos\n",
       "0          1    NaN    NaN  rs11210887  Xu2023   MTAG       EUR  44076019\n",
       "1          1    NaN    NaN   rs2310819  Xu2023   MTAG       EUR  66440096\n",
       "2          1    NaN    NaN  rs11210229  Xu2023   MTAG       EUR  73860028\n",
       "3          1    NaN    NaN   rs1526480  Xu2023   MTAG       EUR  91209986\n",
       "4          1    NaN    NaN   rs3897854  Xu2023   MTAG       EUR  97918306"
      ]
     },
     "execution_count": 34,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "aud_db3.columns = new_column_names1\n",
    "aud_db3.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 35,
   "id": "e3cf5ee1",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Problematic opioid use</th>\n",
       "      <th>Unnamed: 1</th>\n",
       "      <th>Unnamed: 2</th>\n",
       "      <th>Unnamed: 3</th>\n",
       "      <th>Unnamed: 4</th>\n",
       "      <th>Unnamed: 5</th>\n",
       "      <th>Unnamed: 6</th>\n",
       "      <th>Unnamed: 7</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>Date of last update</td>\n",
       "      <td>04.05.2023</td>\n",
       "      <td>KEY</td>\n",
       "      <td>study = abbreviated study name as defined at t...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>Description of search completed</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>method = GWAS/condFDR/conjFDR/MTAG/STAR/other</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>Last update by</td>\n",
       "      <td>Børge</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>Search terms for literature search</td>\n",
       "      <td>Cannabis use disorder + genetic loci</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>EURA = European american</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>List of studies included (author year journal ...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                              Problematic opioid use  \\\n",
       "0                                Date of last update   \n",
       "1                    Description of search completed   \n",
       "2                                     Last update by   \n",
       "3                 Search terms for literature search   \n",
       "4  List of studies included (author year journal ...   \n",
       "\n",
       "                             Unnamed: 1 Unnamed: 2  \\\n",
       "0                            04.05.2023        KEY   \n",
       "1                                   NaN        NaN   \n",
       "2                                 Børge        NaN   \n",
       "3  Cannabis use disorder + genetic loci        NaN   \n",
       "4                                   NaN        NaN   \n",
       "\n",
       "                                          Unnamed: 3 Unnamed: 4 Unnamed: 5  \\\n",
       "0  study = abbreviated study name as defined at t...        NaN        NaN   \n",
       "1      method = GWAS/condFDR/conjFDR/MTAG/STAR/other        NaN        NaN   \n",
       "2                                                NaN        NaN        NaN   \n",
       "3                                                NaN        NaN        NaN   \n",
       "4                                                NaN        NaN        NaN   \n",
       "\n",
       "                 Unnamed: 6 Unnamed: 7  \n",
       "0                       NaN        NaN  \n",
       "1                       NaN        NaN  \n",
       "2                       NaN        NaN  \n",
       "3  EURA = European american        NaN  \n",
       "4                       NaN        NaN  "
      ]
     },
     "execution_count": 35,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "cud_db1 = db1.get('cud')\n",
    "cud_db1.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 37,
   "id": "6f0f0f84",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Problematic opioid use</th>\n",
       "      <th>Unnamed: 1</th>\n",
       "      <th>Unnamed: 2</th>\n",
       "      <th>Unnamed: 3</th>\n",
       "      <th>Unnamed: 4</th>\n",
       "      <th>Unnamed: 5</th>\n",
       "      <th>Unnamed: 6</th>\n",
       "      <th>Unnamed: 7</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chromosome</td>\n",
       "      <td>min_bp</td>\n",
       "      <td>max_bp</td>\n",
       "      <td>lead_SNP</td>\n",
       "      <td>study</td>\n",
       "      <td>method</td>\n",
       "      <td>ethnicity</td>\n",
       "      <td>Pos</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>rs11210887</td>\n",
       "      <td>Xu2023</td>\n",
       "      <td>MTAG</td>\n",
       "      <td>EUR</td>\n",
       "      <td>44076019</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>rs9436444</td>\n",
       "      <td>Xu2023</td>\n",
       "      <td>MTAG</td>\n",
       "      <td>EUR</td>\n",
       "      <td>50589798</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>1</td>\n",
       "      <td>66334537</td>\n",
       "      <td>66385385</td>\n",
       "      <td>rs1937460</td>\n",
       "      <td>Cheng2023</td>\n",
       "      <td>conjFDR</td>\n",
       "      <td>EUR</td>\n",
       "      <td>66334537</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>rs2310819</td>\n",
       "      <td>Xu2023</td>\n",
       "      <td>MTAG</td>\n",
       "      <td>EUR</td>\n",
       "      <td>66440096</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "  Problematic opioid use Unnamed: 1 Unnamed: 2  Unnamed: 3 Unnamed: 4  \\\n",
       "0             chromosome     min_bp     max_bp    lead_SNP      study   \n",
       "1                      1        NaN        NaN  rs11210887     Xu2023   \n",
       "2                      1        NaN        NaN   rs9436444     Xu2023   \n",
       "3                      1   66334537   66385385   rs1937460  Cheng2023   \n",
       "4                      1        NaN        NaN   rs2310819     Xu2023   \n",
       "\n",
       "  Unnamed: 5 Unnamed: 6 Unnamed: 7  \n",
       "0     method  ethnicity        Pos  \n",
       "1       MTAG        EUR   44076019  \n",
       "2       MTAG        EUR   50589798  \n",
       "3    conjFDR        EUR   66334537  \n",
       "4       MTAG        EUR   66440096  "
      ]
     },
     "execution_count": 37,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "cud_db2=cud_db1.iloc[8:].reset_index(drop=True)\n",
    "cud_db2.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 38,
   "id": "942ecee4",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Problematic opioid use    chromosome\n",
       "Unnamed: 1                    min_bp\n",
       "Unnamed: 2                    max_bp\n",
       "Unnamed: 3                  lead_SNP\n",
       "Unnamed: 4                     study\n",
       "Unnamed: 5                    method\n",
       "Unnamed: 6                 ethnicity\n",
       "Unnamed: 7                       Pos\n",
       "Name: 0, dtype: object"
      ]
     },
     "execution_count": 38,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "new_column_names2 = cud_db2.iloc[0]\n",
    "new_column_names2"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 39,
   "id": "28907a24",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Problematic opioid use</th>\n",
       "      <th>Unnamed: 1</th>\n",
       "      <th>Unnamed: 2</th>\n",
       "      <th>Unnamed: 3</th>\n",
       "      <th>Unnamed: 4</th>\n",
       "      <th>Unnamed: 5</th>\n",
       "      <th>Unnamed: 6</th>\n",
       "      <th>Unnamed: 7</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>rs11210887</td>\n",
       "      <td>Xu2023</td>\n",
       "      <td>MTAG</td>\n",
       "      <td>EUR</td>\n",
       "      <td>44076019</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>rs9436444</td>\n",
       "      <td>Xu2023</td>\n",
       "      <td>MTAG</td>\n",
       "      <td>EUR</td>\n",
       "      <td>50589798</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>1</td>\n",
       "      <td>66334537</td>\n",
       "      <td>66385385</td>\n",
       "      <td>rs1937460</td>\n",
       "      <td>Cheng2023</td>\n",
       "      <td>conjFDR</td>\n",
       "      <td>EUR</td>\n",
       "      <td>66334537</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>rs2310819</td>\n",
       "      <td>Xu2023</td>\n",
       "      <td>MTAG</td>\n",
       "      <td>EUR</td>\n",
       "      <td>66440096</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>rs4593766</td>\n",
       "      <td>Xu2023</td>\n",
       "      <td>MTAG</td>\n",
       "      <td>EUR</td>\n",
       "      <td>73773043</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "  Problematic opioid use Unnamed: 1 Unnamed: 2  Unnamed: 3 Unnamed: 4  \\\n",
       "0                      1        NaN        NaN  rs11210887     Xu2023   \n",
       "1                      1        NaN        NaN   rs9436444     Xu2023   \n",
       "2                      1   66334537   66385385   rs1937460  Cheng2023   \n",
       "3                      1        NaN        NaN   rs2310819     Xu2023   \n",
       "4                      1        NaN        NaN   rs4593766     Xu2023   \n",
       "\n",
       "  Unnamed: 5 Unnamed: 6 Unnamed: 7  \n",
       "0       MTAG        EUR   44076019  \n",
       "1       MTAG        EUR   50589798  \n",
       "2    conjFDR        EUR   66334537  \n",
       "3       MTAG        EUR   66440096  \n",
       "4       MTAG        EUR   73773043  "
      ]
     },
     "execution_count": 39,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "cud_db3 = cud_db2[1:].reset_index(drop=True)\n",
    "cud_db3.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 40,
   "id": "fb9ce0eb",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>chromosome</th>\n",
       "      <th>min_bp</th>\n",
       "      <th>max_bp</th>\n",
       "      <th>lead_SNP</th>\n",
       "      <th>study</th>\n",
       "      <th>method</th>\n",
       "      <th>ethnicity</th>\n",
       "      <th>Pos</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>rs11210887</td>\n",
       "      <td>Xu2023</td>\n",
       "      <td>MTAG</td>\n",
       "      <td>EUR</td>\n",
       "      <td>44076019</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>rs9436444</td>\n",
       "      <td>Xu2023</td>\n",
       "      <td>MTAG</td>\n",
       "      <td>EUR</td>\n",
       "      <td>50589798</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>1</td>\n",
       "      <td>66334537</td>\n",
       "      <td>66385385</td>\n",
       "      <td>rs1937460</td>\n",
       "      <td>Cheng2023</td>\n",
       "      <td>conjFDR</td>\n",
       "      <td>EUR</td>\n",
       "      <td>66334537</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>rs2310819</td>\n",
       "      <td>Xu2023</td>\n",
       "      <td>MTAG</td>\n",
       "      <td>EUR</td>\n",
       "      <td>66440096</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>rs4593766</td>\n",
       "      <td>Xu2023</td>\n",
       "      <td>MTAG</td>\n",
       "      <td>EUR</td>\n",
       "      <td>73773043</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "0 chromosome    min_bp    max_bp    lead_SNP      study   method ethnicity  \\\n",
       "0          1       NaN       NaN  rs11210887     Xu2023     MTAG       EUR   \n",
       "1          1       NaN       NaN   rs9436444     Xu2023     MTAG       EUR   \n",
       "2          1  66334537  66385385   rs1937460  Cheng2023  conjFDR       EUR   \n",
       "3          1       NaN       NaN   rs2310819     Xu2023     MTAG       EUR   \n",
       "4          1       NaN       NaN   rs4593766     Xu2023     MTAG       EUR   \n",
       "\n",
       "0       Pos  \n",
       "0  44076019  \n",
       "1  50589798  \n",
       "2  66334537  \n",
       "3  66440096  \n",
       "4  73773043  "
      ]
     },
     "execution_count": 40,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "cud_db3.columns = new_column_names2\n",
    "cud_db3.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 41,
   "id": "f5da0542",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Opioid use disorder</th>\n",
       "      <th>Unnamed: 1</th>\n",
       "      <th>Unnamed: 2</th>\n",
       "      <th>Unnamed: 3</th>\n",
       "      <th>Unnamed: 4</th>\n",
       "      <th>Unnamed: 5</th>\n",
       "      <th>Unnamed: 6</th>\n",
       "      <th>Unnamed: 7</th>\n",
       "      <th>Unnamed: 8</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>Date of last update</td>\n",
       "      <td>03.05.2023</td>\n",
       "      <td>KEY</td>\n",
       "      <td>study = abbreviated study name as defined at t...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>Description of search completed</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>method = GWAS/condFDR/conjFDR/MTAG/STAR/other</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>Last update by</td>\n",
       "      <td>Børge</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>Search terms for literature search</td>\n",
       "      <td>Opioid use disorder + genetic loci</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>EURA = European american</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>List of studies included (author year journal ...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>AFRA = African american</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                 Opioid use disorder  \\\n",
       "0                                Date of last update   \n",
       "1                    Description of search completed   \n",
       "2                                     Last update by   \n",
       "3                 Search terms for literature search   \n",
       "4  List of studies included (author year journal ...   \n",
       "\n",
       "                           Unnamed: 1 Unnamed: 2  \\\n",
       "0                          03.05.2023        KEY   \n",
       "1                                 NaN        NaN   \n",
       "2                               Børge        NaN   \n",
       "3  Opioid use disorder + genetic loci        NaN   \n",
       "4                                 NaN        NaN   \n",
       "\n",
       "                                          Unnamed: 3 Unnamed: 4 Unnamed: 5  \\\n",
       "0  study = abbreviated study name as defined at t...        NaN        NaN   \n",
       "1      method = GWAS/condFDR/conjFDR/MTAG/STAR/other        NaN        NaN   \n",
       "2                                                NaN        NaN        NaN   \n",
       "3                                                NaN        NaN        NaN   \n",
       "4                                                NaN        NaN        NaN   \n",
       "\n",
       "                 Unnamed: 6 Unnamed: 7 Unnamed: 8  \n",
       "0                       NaN        NaN        NaN  \n",
       "1                       NaN        NaN        NaN  \n",
       "2                       NaN        NaN        NaN  \n",
       "3  EURA = European american        NaN        NaN  \n",
       "4   AFRA = African american        NaN        NaN  "
      ]
     },
     "execution_count": 41,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "oud_db1 = db1.get('oud')\n",
    "oud_db1.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 43,
   "id": "7516fa5a",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Opioid use disorder</th>\n",
       "      <th>Unnamed: 1</th>\n",
       "      <th>Unnamed: 2</th>\n",
       "      <th>Unnamed: 3</th>\n",
       "      <th>Unnamed: 4</th>\n",
       "      <th>Unnamed: 5</th>\n",
       "      <th>Unnamed: 6</th>\n",
       "      <th>Unnamed: 7</th>\n",
       "      <th>Unnamed: 8</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chromosome</td>\n",
       "      <td>min_bp</td>\n",
       "      <td>max_bp</td>\n",
       "      <td>lead_SNP</td>\n",
       "      <td>study</td>\n",
       "      <td>method</td>\n",
       "      <td>ethnicity</td>\n",
       "      <td>Pos</td>\n",
       "      <td>Association with other substance use trait</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>1</td>\n",
       "      <td>27989020</td>\n",
       "      <td>29989020</td>\n",
       "      <td>rs6667501</td>\n",
       "      <td>Deak2021</td>\n",
       "      <td>MTAG</td>\n",
       "      <td>EUR</td>\n",
       "      <td>28989020</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>1</td>\n",
       "      <td>55522425</td>\n",
       "      <td>55708610</td>\n",
       "      <td>rs67237321</td>\n",
       "      <td>Holen2023</td>\n",
       "      <td>condFDR</td>\n",
       "      <td>EUR</td>\n",
       "      <td>55686228</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>1</td>\n",
       "      <td>65434743</td>\n",
       "      <td>67434743</td>\n",
       "      <td>rs7519259</td>\n",
       "      <td>Deak2021</td>\n",
       "      <td>MTAG</td>\n",
       "      <td>EUR</td>\n",
       "      <td>66434743</td>\n",
       "      <td>Smoking, Alcohol (consumption and problematic ...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>1</td>\n",
       "      <td>72860028</td>\n",
       "      <td>74860028</td>\n",
       "      <td>rs11210229</td>\n",
       "      <td>Xu2022</td>\n",
       "      <td>MTAG</td>\n",
       "      <td>EUR</td>\n",
       "      <td>73860028</td>\n",
       "      <td>Smoking, Alcohol (consumption, problematic use...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "  Opioid use disorder Unnamed: 1 Unnamed: 2  Unnamed: 3 Unnamed: 4 Unnamed: 5  \\\n",
       "0          chromosome     min_bp     max_bp    lead_SNP      study     method   \n",
       "1                   1   27989020   29989020   rs6667501   Deak2021       MTAG   \n",
       "2                   1   55522425   55708610  rs67237321  Holen2023    condFDR   \n",
       "3                   1   65434743   67434743   rs7519259   Deak2021       MTAG   \n",
       "4                   1   72860028   74860028  rs11210229     Xu2022       MTAG   \n",
       "\n",
       "  Unnamed: 6 Unnamed: 7                                         Unnamed: 8  \n",
       "0  ethnicity        Pos         Association with other substance use trait  \n",
       "1        EUR   28989020                                                NaN  \n",
       "2        EUR   55686228                                                NaN  \n",
       "3        EUR   66434743  Smoking, Alcohol (consumption and problematic ...  \n",
       "4        EUR   73860028  Smoking, Alcohol (consumption, problematic use...  "
      ]
     },
     "execution_count": 43,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "oud_db2=oud_db1.iloc[10:].reset_index(drop=True)\n",
    "oud_db2.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 44,
   "id": "24385283",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Opioid use disorder                                    chromosome\n",
       "Unnamed: 1                                                 min_bp\n",
       "Unnamed: 2                                                 max_bp\n",
       "Unnamed: 3                                               lead_SNP\n",
       "Unnamed: 4                                                  study\n",
       "Unnamed: 5                                                 method\n",
       "Unnamed: 6                                              ethnicity\n",
       "Unnamed: 7                                                    Pos\n",
       "Unnamed: 8             Association with other substance use trait\n",
       "Name: 0, dtype: object"
      ]
     },
     "execution_count": 44,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "new_column_names3 = oud_db2.iloc[0]\n",
    "new_column_names3"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 45,
   "id": "c07cfece",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Opioid use disorder</th>\n",
       "      <th>Unnamed: 1</th>\n",
       "      <th>Unnamed: 2</th>\n",
       "      <th>Unnamed: 3</th>\n",
       "      <th>Unnamed: 4</th>\n",
       "      <th>Unnamed: 5</th>\n",
       "      <th>Unnamed: 6</th>\n",
       "      <th>Unnamed: 7</th>\n",
       "      <th>Unnamed: 8</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1</td>\n",
       "      <td>27989020</td>\n",
       "      <td>29989020</td>\n",
       "      <td>rs6667501</td>\n",
       "      <td>Deak2021</td>\n",
       "      <td>MTAG</td>\n",
       "      <td>EUR</td>\n",
       "      <td>28989020</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>1</td>\n",
       "      <td>55522425</td>\n",
       "      <td>55708610</td>\n",
       "      <td>rs67237321</td>\n",
       "      <td>Holen2023</td>\n",
       "      <td>condFDR</td>\n",
       "      <td>EUR</td>\n",
       "      <td>55686228</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>1</td>\n",
       "      <td>65434743</td>\n",
       "      <td>67434743</td>\n",
       "      <td>rs7519259</td>\n",
       "      <td>Deak2021</td>\n",
       "      <td>MTAG</td>\n",
       "      <td>EUR</td>\n",
       "      <td>66434743</td>\n",
       "      <td>Smoking, Alcohol (consumption and problematic ...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>1</td>\n",
       "      <td>72860028</td>\n",
       "      <td>74860028</td>\n",
       "      <td>rs11210229</td>\n",
       "      <td>Xu2022</td>\n",
       "      <td>MTAG</td>\n",
       "      <td>EUR</td>\n",
       "      <td>73860028</td>\n",
       "      <td>Smoking, Alcohol (consumption, problematic use...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>1</td>\n",
       "      <td>90208451</td>\n",
       "      <td>92208451</td>\n",
       "      <td>rs2166171</td>\n",
       "      <td>Deak2021</td>\n",
       "      <td>MTAG</td>\n",
       "      <td>EUR</td>\n",
       "      <td>91208451</td>\n",
       "      <td>Smoking</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "  Opioid use disorder Unnamed: 1 Unnamed: 2  Unnamed: 3 Unnamed: 4 Unnamed: 5  \\\n",
       "0                   1   27989020   29989020   rs6667501   Deak2021       MTAG   \n",
       "1                   1   55522425   55708610  rs67237321  Holen2023    condFDR   \n",
       "2                   1   65434743   67434743   rs7519259   Deak2021       MTAG   \n",
       "3                   1   72860028   74860028  rs11210229     Xu2022       MTAG   \n",
       "4                   1   90208451   92208451   rs2166171   Deak2021       MTAG   \n",
       "\n",
       "  Unnamed: 6 Unnamed: 7                                         Unnamed: 8  \n",
       "0        EUR   28989020                                                NaN  \n",
       "1        EUR   55686228                                                NaN  \n",
       "2        EUR   66434743  Smoking, Alcohol (consumption and problematic ...  \n",
       "3        EUR   73860028  Smoking, Alcohol (consumption, problematic use...  \n",
       "4        EUR   91208451                                            Smoking  "
      ]
     },
     "execution_count": 45,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "oud_db3 = oud_db2[1:].reset_index(drop=True)\n",
    "oud_db3.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 46,
   "id": "b42de94f",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>chromosome</th>\n",
       "      <th>min_bp</th>\n",
       "      <th>max_bp</th>\n",
       "      <th>lead_SNP</th>\n",
       "      <th>study</th>\n",
       "      <th>method</th>\n",
       "      <th>ethnicity</th>\n",
       "      <th>Pos</th>\n",
       "      <th>Association with other substance use trait</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1</td>\n",
       "      <td>27989020</td>\n",
       "      <td>29989020</td>\n",
       "      <td>rs6667501</td>\n",
       "      <td>Deak2021</td>\n",
       "      <td>MTAG</td>\n",
       "      <td>EUR</td>\n",
       "      <td>28989020</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>1</td>\n",
       "      <td>55522425</td>\n",
       "      <td>55708610</td>\n",
       "      <td>rs67237321</td>\n",
       "      <td>Holen2023</td>\n",
       "      <td>condFDR</td>\n",
       "      <td>EUR</td>\n",
       "      <td>55686228</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>1</td>\n",
       "      <td>65434743</td>\n",
       "      <td>67434743</td>\n",
       "      <td>rs7519259</td>\n",
       "      <td>Deak2021</td>\n",
       "      <td>MTAG</td>\n",
       "      <td>EUR</td>\n",
       "      <td>66434743</td>\n",
       "      <td>Smoking, Alcohol (consumption and problematic ...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>1</td>\n",
       "      <td>72860028</td>\n",
       "      <td>74860028</td>\n",
       "      <td>rs11210229</td>\n",
       "      <td>Xu2022</td>\n",
       "      <td>MTAG</td>\n",
       "      <td>EUR</td>\n",
       "      <td>73860028</td>\n",
       "      <td>Smoking, Alcohol (consumption, problematic use...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>1</td>\n",
       "      <td>90208451</td>\n",
       "      <td>92208451</td>\n",
       "      <td>rs2166171</td>\n",
       "      <td>Deak2021</td>\n",
       "      <td>MTAG</td>\n",
       "      <td>EUR</td>\n",
       "      <td>91208451</td>\n",
       "      <td>Smoking</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "0 chromosome    min_bp    max_bp    lead_SNP      study   method ethnicity  \\\n",
       "0          1  27989020  29989020   rs6667501   Deak2021     MTAG       EUR   \n",
       "1          1  55522425  55708610  rs67237321  Holen2023  condFDR       EUR   \n",
       "2          1  65434743  67434743   rs7519259   Deak2021     MTAG       EUR   \n",
       "3          1  72860028  74860028  rs11210229     Xu2022     MTAG       EUR   \n",
       "4          1  90208451  92208451   rs2166171   Deak2021     MTAG       EUR   \n",
       "\n",
       "0       Pos         Association with other substance use trait  \n",
       "0  28989020                                                NaN  \n",
       "1  55686228                                                NaN  \n",
       "2  66434743  Smoking, Alcohol (consumption and problematic ...  \n",
       "3  73860028  Smoking, Alcohol (consumption, problematic use...  \n",
       "4  91208451                                            Smoking  "
      ]
     },
     "execution_count": 46,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "oud_db3.columns = new_column_names3\n",
    "oud_db3.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 48,
   "id": "9f414e14",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>chromosome</th>\n",
       "      <th>min_bp</th>\n",
       "      <th>max_bp</th>\n",
       "      <th>lead_SNP</th>\n",
       "      <th>study</th>\n",
       "      <th>method</th>\n",
       "      <th>ethnicity</th>\n",
       "      <th>Pos</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>rs11210887</td>\n",
       "      <td>Xu2023</td>\n",
       "      <td>MTAG</td>\n",
       "      <td>EUR</td>\n",
       "      <td>44076019</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>rs2310819</td>\n",
       "      <td>Xu2023</td>\n",
       "      <td>MTAG</td>\n",
       "      <td>EUR</td>\n",
       "      <td>66440096</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>rs11210229</td>\n",
       "      <td>Xu2023</td>\n",
       "      <td>MTAG</td>\n",
       "      <td>EUR</td>\n",
       "      <td>73860028</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>rs1526480</td>\n",
       "      <td>Xu2023</td>\n",
       "      <td>MTAG</td>\n",
       "      <td>EUR</td>\n",
       "      <td>91209986</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>rs3897854</td>\n",
       "      <td>Xu2023</td>\n",
       "      <td>MTAG</td>\n",
       "      <td>EUR</td>\n",
       "      <td>97918306</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "0 chromosome min_bp max_bp    lead_SNP   study method ethnicity       Pos\n",
       "0          1    NaN    NaN  rs11210887  Xu2023   MTAG       EUR  44076019\n",
       "1          1    NaN    NaN   rs2310819  Xu2023   MTAG       EUR  66440096\n",
       "2          1    NaN    NaN  rs11210229  Xu2023   MTAG       EUR  73860028\n",
       "3          1    NaN    NaN   rs1526480  Xu2023   MTAG       EUR  91209986\n",
       "4          1    NaN    NaN   rs3897854  Xu2023   MTAG       EUR  97918306"
      ]
     },
     "execution_count": 48,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "aud_db3.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 49,
   "id": "f5dd06e2",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>chromosome</th>\n",
       "      <th>min_bp</th>\n",
       "      <th>max_bp</th>\n",
       "      <th>lead_SNP</th>\n",
       "      <th>study</th>\n",
       "      <th>method</th>\n",
       "      <th>ethnicity</th>\n",
       "      <th>Pos</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>rs11210887</td>\n",
       "      <td>Xu2023</td>\n",
       "      <td>MTAG</td>\n",
       "      <td>EUR</td>\n",
       "      <td>44076019</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>rs9436444</td>\n",
       "      <td>Xu2023</td>\n",
       "      <td>MTAG</td>\n",
       "      <td>EUR</td>\n",
       "      <td>50589798</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>1</td>\n",
       "      <td>66334537</td>\n",
       "      <td>66385385</td>\n",
       "      <td>rs1937460</td>\n",
       "      <td>Cheng2023</td>\n",
       "      <td>conjFDR</td>\n",
       "      <td>EUR</td>\n",
       "      <td>66334537</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>rs2310819</td>\n",
       "      <td>Xu2023</td>\n",
       "      <td>MTAG</td>\n",
       "      <td>EUR</td>\n",
       "      <td>66440096</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>rs4593766</td>\n",
       "      <td>Xu2023</td>\n",
       "      <td>MTAG</td>\n",
       "      <td>EUR</td>\n",
       "      <td>73773043</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "0 chromosome    min_bp    max_bp    lead_SNP      study   method ethnicity  \\\n",
       "0          1       NaN       NaN  rs11210887     Xu2023     MTAG       EUR   \n",
       "1          1       NaN       NaN   rs9436444     Xu2023     MTAG       EUR   \n",
       "2          1  66334537  66385385   rs1937460  Cheng2023  conjFDR       EUR   \n",
       "3          1       NaN       NaN   rs2310819     Xu2023     MTAG       EUR   \n",
       "4          1       NaN       NaN   rs4593766     Xu2023     MTAG       EUR   \n",
       "\n",
       "0       Pos  \n",
       "0  44076019  \n",
       "1  50589798  \n",
       "2  66334537  \n",
       "3  66440096  \n",
       "4  73773043  "
      ]
     },
     "execution_count": 49,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "cud_db3.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 50,
   "id": "58debcc9",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "dict_keys(['CUD|OUD', 'CUD|AUD', 'OUD|CUD', 'OUD|AUD', 'AUD|CUD', 'AUD|OUD'])"
      ]
     },
     "execution_count": 50,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "condfdr1 = pd.read_excel('fdr_out/condfdr_lead_snp2gene_effects_fin1_novelty.xlsx',sheet_name=None)\n",
    "condfdr1.keys()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 76,
   "id": "4287ae43",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "CUD (24, 3) OUD (31, 3)    locusnum  CHR    LEAD_SNP    LEAD_BP      MinBP      MaxBP       FDR  \\\n",
      "0         1    2  rs12616949   22594027   22549441   22606275  0.006116   \n",
      "1         2    5  rs12513581   60533207   60484179   60614879  0.008724   \n",
      "2         3    6    rs975303   19028788   19023726   19191459  0.001397   \n",
      "3         4    6   rs3734540   26463321   26393021   26581168  0.001896   \n",
      "4         5    7   rs1989903  114137940  114015707  114287116  0.001041   \n",
      "\n",
      "   is_locus_lead A1 A2  ...  PVAL_in_OUD             Gene Approved symbol  \\\n",
      "0           True  A  G  ...     0.003622              NaN             NaN   \n",
      "1           True  C  T  ...     0.002643  ENSG00000164181          ELOVL7   \n",
      "2           True  G  A  ...     0.000019              NaN             NaN   \n",
      "3           True  C  A  ...     0.002091  ENSG00000182952           HMGN4   \n",
      "4           True  A  G  ...     0.082183  ENSG00000128573           FOXP2   \n",
      "\n",
      "                                      Approved name  Overall_Gene_Score  \\\n",
      "0                                               NaN                 NaN   \n",
      "1                       ELOVL fatty acid elongase 7            0.214085   \n",
      "2                                               NaN                 NaN   \n",
      "3  high mobility group nucleosomal binding domain 4            0.426962   \n",
      "4                                   forkhead box P2            0.033400   \n",
      "\n",
      "  Novel_leadsnp_db1 Novel_leadsnp_db2 Novel_loci_in_db1  Novel_loci_in_db2  \\\n",
      "0               Yes               Yes                No                Yes   \n",
      "1               Yes               Yes                No                 No   \n",
      "2               Yes               Yes               Yes                Yes   \n",
      "3                No               Yes                No                 No   \n",
      "4               Yes               Yes                No                Yes   \n",
      "\n",
      "  Novel_loci  \n",
      "0         No  \n",
      "1         No  \n",
      "2        Yes  \n",
      "3         No  \n",
      "4         No  \n",
      "\n",
      "[5 rows x 24 columns]\n",
      "CUD (24, 3) AUD (4, 3)    locusnum  CHR    LEAD_SNP   LEAD_BP     MinBP     MaxBP       FDR  \\\n",
      "0         1    1   rs1392816  66481188  66334537  66485442  0.000036   \n",
      "1         2    1  rs12122743  91220791  91200757  91234126  0.006585   \n",
      "2         3    5    rs173946  60488297  60135962  60550041  0.001336   \n",
      "3         4    5   rs6452787  87712831  87514778  87822672  0.003577   \n",
      "4         5    6  rs72828558  19128832  19003340  19191459  0.003909   \n",
      "\n",
      "   is_locus_lead A1 A2  ...  PVAL_in_AUD             Gene Approved symbol  \\\n",
      "0           True  T  C  ...     0.000317  ENSG00000213625          LEPROT   \n",
      "1           True  G  A  ...     0.003699  ENSG00000143032          BARHL2   \n",
      "2           True  C  T  ...     0.001329  ENSG00000164181          ELOVL7   \n",
      "3           True  G  A  ...     0.003701  ENSG00000164180        TMEM161B   \n",
      "4           True  C  T  ...     0.000126              NaN             NaN   \n",
      "\n",
      "                            Approved name  Overall_Gene_Score  \\\n",
      "0  leptin receptor overlapping transcript            0.079678   \n",
      "1                    BarH like homeobox 2            0.059759   \n",
      "2             ELOVL fatty acid elongase 7            0.240845   \n",
      "3              transmembrane protein 161B            0.046479   \n",
      "4                                     NaN                 NaN   \n",
      "\n",
      "  Novel_leadsnp_db1 Novel_leadsnp_db2 Novel_loci_in_db1  Novel_loci_in_db2  \\\n",
      "0               Yes               Yes                No                 No   \n",
      "1               Yes               Yes                No                Yes   \n",
      "2               Yes               Yes               Yes                Yes   \n",
      "3               Yes               Yes               Yes                Yes   \n",
      "4               Yes               Yes               Yes                Yes   \n",
      "\n",
      "  Novel_loci  \n",
      "0         No  \n",
      "1         No  \n",
      "2        Yes  \n",
      "3        Yes  \n",
      "4        Yes  \n",
      "\n",
      "[5 rows x 24 columns]\n",
      "OUD (31, 3) CUD (24, 3)    locusnum  CHR     LEAD_SNP    LEAD_BP      MinBP      MaxBP       FDR  \\\n",
      "0         1    3  rs199698239  107981430  107880069  108031094  0.004063   \n",
      "1         2    6   rs72828517   19036035   19023726   19191459  0.006021   \n",
      "2         3    6   rs35984974   27410422   26175866   28397676  0.002924   \n",
      "3         4    6    rs1799971  154360797  154194081  154390607  0.000012   \n",
      "4         5    9  rs185062166  127961612  127780246  127987802  0.004438   \n",
      "\n",
      "   is_locus_lead A1 A2  ...   PVAL_in_CUD             Gene Approved symbol  \\\n",
      "0           True  A  G  ...  9.093000e-01  ENSG00000114455           HHLA2   \n",
      "1           True  C  T  ...  1.860000e-03  ENSG00000137393         RNF144B   \n",
      "2           True  A  G  ...  9.471000e-08  ENSG00000112812          PRSS16   \n",
      "3           True  G  A  ...  1.594000e-01  ENSG00000112038           OPRM1   \n",
      "4           True  A  G  ...  9.215000e-10  ENSG00000136933          RABEPK   \n",
      "\n",
      "                             Approved name  Overall_Gene_Score  \\\n",
      "0                HHLA2 member of B7 family            0.253320   \n",
      "1                 ring finger protein 144B            0.000000   \n",
      "2                       serine protease 16            0.132797   \n",
      "3                     opioid receptor mu 1            0.205835   \n",
      "4  Rab9 effector protein with kelch motifs            0.199195   \n",
      "\n",
      "  Novel_leadsnp_db1 Novel_leadsnp_db2 Novel_loci_in_db1  Novel_loci_in_db2  \\\n",
      "0               Yes               Yes               Yes                Yes   \n",
      "1               Yes               Yes               Yes                Yes   \n",
      "2               Yes               Yes                No                 No   \n",
      "3               Yes               Yes               Yes                Yes   \n",
      "4               Yes               Yes                No                Yes   \n",
      "\n",
      "  Novel_loci  \n",
      "0        Yes  \n",
      "1        Yes  \n",
      "2         No  \n",
      "3        Yes  \n",
      "4         No  \n",
      "\n",
      "[5 rows x 24 columns]\n",
      "OUD (31, 3) AUD (4, 3)    locusnum  CHR    LEAD_SNP    LEAD_BP      MinBP      MaxBP           FDR  \\\n",
      "0         1    1  rs12031155   53714139   53658317   53752134  9.997570e-03   \n",
      "1         2    2  rs35942385  144208523  144145478  144263280  2.498582e-03   \n",
      "2         3    6  rs72828517   19036035   19023726   19191459  7.457391e-03   \n",
      "3         4    6   rs3094551   29354799   28674322   29354809  2.603033e-04   \n",
      "4         5    6   rs1799971  154360797  154194081  154390607  2.066166e-08   \n",
      "\n",
      "   is_locus_lead A1 A2  ...   PVAL_in_AUD             Gene Approved symbol  \\\n",
      "0           True  T  C  ...  1.683000e-03  ENSG00000157184            CPT2   \n",
      "1           True  T  G  ...  2.418000e-08  ENSG00000075884        ARHGAP15   \n",
      "2           True  C  T  ...  8.597000e-03  ENSG00000137393         RNF144B   \n",
      "3           True  A  G  ...  1.660000e-02  ENSG00000204644           ZFP57   \n",
      "4           True  G  A  ...  1.502000e-05  ENSG00000112038           OPRM1   \n",
      "\n",
      "                      Approved name  Overall_Gene_Score Novel_leadsnp_db1  \\\n",
      "0  carnitine palmitoyltransferase 2            0.213481               Yes   \n",
      "1  Rho GTPase activating protein 15            0.146278                No   \n",
      "2          ring finger protein 144B            0.000000               Yes   \n",
      "3         ZFP57 zinc finger protein            0.119517               Yes   \n",
      "4              opioid receptor mu 1            0.205835               Yes   \n",
      "\n",
      "  Novel_leadsnp_db2 Novel_loci_in_db1  Novel_loci_in_db2 Novel_loci  \n",
      "0               Yes               Yes                Yes        Yes  \n",
      "1               Yes                No                Yes         No  \n",
      "2               Yes               Yes                Yes        Yes  \n",
      "3               Yes                No                Yes         No  \n",
      "4               Yes               Yes                Yes        Yes  \n",
      "\n",
      "[5 rows x 24 columns]\n",
      "AUD (4, 3) CUD (24, 3)    locusnum  CHR    LEAD_SNP    LEAD_BP      MinBP      MaxBP       FDR  \\\n",
      "0         1    1   rs2310819   66440096   66334518   66547212  0.000001   \n",
      "1         2    1  rs12145432   73769330   73766431   73991651  0.000226   \n",
      "2         3    1  rs61535100   80827756   80784642   80871734  0.004921   \n",
      "3         4    1   rs2166171   91208451   91189731   91234126  0.000117   \n",
      "4         5    1   rs7544174  236902955  236897248  236905585  0.003473   \n",
      "\n",
      "   is_locus_lead A1 A2  ...   PVAL_in_CUD             Gene Approved symbol  \\\n",
      "0           True  C  T  ...  2.530000e-09  ENSG00000213625          LEPROT   \n",
      "1           True  T  G  ...  1.107000e-07              NaN             NaN   \n",
      "2           True  G  A  ...  5.544000e-05              NaN             NaN   \n",
      "3           True  C  T  ...  8.294000e-07  ENSG00000143032          BARHL2   \n",
      "4           True  T  C  ...  1.419000e-02  ENSG00000077522           ACTN2   \n",
      "\n",
      "                            Approved name  Overall_Gene_Score  \\\n",
      "0  leptin receptor overlapping transcript            0.066398   \n",
      "1                                     NaN                 NaN   \n",
      "2                                     NaN                 NaN   \n",
      "3                    BarH like homeobox 2            0.059759   \n",
      "4                         actinin alpha 2            0.166600   \n",
      "\n",
      "  Novel_leadsnp_db1 Novel_leadsnp_db2 Novel_loci_in_db1  Novel_loci_in_db2  \\\n",
      "0               Yes               Yes                No                 No   \n",
      "1               Yes               Yes                No                Yes   \n",
      "2               Yes               Yes               Yes                Yes   \n",
      "3               Yes               Yes               Yes                 No   \n",
      "4               Yes               Yes               Yes                Yes   \n",
      "\n",
      "  Novel_loci  \n",
      "0         No  \n",
      "1         No  \n",
      "2        Yes  \n",
      "3         No  \n",
      "4        Yes  \n",
      "\n",
      "[5 rows x 24 columns]\n",
      "AUD (4, 3) OUD (31, 3)    locusnum  CHR    LEAD_SNP    LEAD_BP      MinBP      MaxBP       FDR  \\\n",
      "0         1    4  rs13125440   39422324   39413780   39425248  0.006060   \n",
      "1         2    4   rs2213041  100247351  100246711  100291659  0.000003   \n",
      "2         3    5    rs331757  124844266  124820663  124852240  0.001326   \n",
      "3         4    6   rs7757112  154151020  153995311  154390607  0.003123   \n",
      "4         5    7   rs2551777  135100476  135050259  135221170  0.000006   \n",
      "\n",
      "   is_locus_lead A1 A2  ...  PVAL_in_OUD             Gene Approved symbol  \\\n",
      "0           True  G  A  ...     0.472779  ENSG00000109814            UGDH   \n",
      "1           True  A  C  ...     0.136588  ENSG00000248144           ADH1C   \n",
      "2           True  G  A  ...     0.052753              NaN             NaN   \n",
      "3           True  C  T  ...     0.001247  ENSG00000112038           OPRM1   \n",
      "4           True  C  T  ...     0.000046  ENSG00000243317           STMP1   \n",
      "\n",
      "                                       Approved name  Overall_Gene_Score  \\\n",
      "0                        UDP-glucose 6-dehydrogenase            0.172636   \n",
      "1  alcohol dehydrogenase 1C (class I), gamma poly...            0.199799   \n",
      "2                                                NaN                 NaN   \n",
      "3                               opioid receptor mu 1            0.039839   \n",
      "4        short transmembrane mitochondrial protein 1            0.165996   \n",
      "\n",
      "  Novel_leadsnp_db1 Novel_leadsnp_db2 Novel_loci_in_db1  Novel_loci_in_db2  \\\n",
      "0               Yes               Yes               Yes                Yes   \n",
      "1               Yes               Yes               Yes                Yes   \n",
      "2               Yes               Yes               Yes                Yes   \n",
      "3               Yes               Yes               Yes                Yes   \n",
      "4               Yes               Yes               Yes                Yes   \n",
      "\n",
      "  Novel_loci  \n",
      "0        Yes  \n",
      "1        Yes  \n",
      "2        Yes  \n",
      "3        Yes  \n",
      "4        Yes  \n",
      "\n",
      "[5 rows x 24 columns]\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/var/folders/g7/p8558cx50s5gdqg8zvqrdf1h0000gn/T/ipykernel_1874/2763236127.py:28: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  db1.dropna(inplace=True)\n",
      "/var/folders/g7/p8558cx50s5gdqg8zvqrdf1h0000gn/T/ipykernel_1874/2763236127.py:29: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  db2.dropna(inplace=True)\n",
      "/var/folders/g7/p8558cx50s5gdqg8zvqrdf1h0000gn/T/ipykernel_1874/2763236127.py:28: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  db1.dropna(inplace=True)\n",
      "/var/folders/g7/p8558cx50s5gdqg8zvqrdf1h0000gn/T/ipykernel_1874/2763236127.py:29: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  db2.dropna(inplace=True)\n",
      "/var/folders/g7/p8558cx50s5gdqg8zvqrdf1h0000gn/T/ipykernel_1874/2763236127.py:28: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  db1.dropna(inplace=True)\n",
      "/var/folders/g7/p8558cx50s5gdqg8zvqrdf1h0000gn/T/ipykernel_1874/2763236127.py:29: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  db2.dropna(inplace=True)\n",
      "/var/folders/g7/p8558cx50s5gdqg8zvqrdf1h0000gn/T/ipykernel_1874/2763236127.py:28: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  db1.dropna(inplace=True)\n",
      "/var/folders/g7/p8558cx50s5gdqg8zvqrdf1h0000gn/T/ipykernel_1874/2763236127.py:29: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  db2.dropna(inplace=True)\n",
      "/var/folders/g7/p8558cx50s5gdqg8zvqrdf1h0000gn/T/ipykernel_1874/2763236127.py:28: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  db1.dropna(inplace=True)\n",
      "/var/folders/g7/p8558cx50s5gdqg8zvqrdf1h0000gn/T/ipykernel_1874/2763236127.py:29: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  db2.dropna(inplace=True)\n",
      "/var/folders/g7/p8558cx50s5gdqg8zvqrdf1h0000gn/T/ipykernel_1874/2763236127.py:28: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  db1.dropna(inplace=True)\n",
      "/var/folders/g7/p8558cx50s5gdqg8zvqrdf1h0000gn/T/ipykernel_1874/2763236127.py:29: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  db2.dropna(inplace=True)\n"
     ]
    }
   ],
   "source": [
    "w2 = ExcelWriter('fdr_out/condfdr_lead_snp2gene_effects_fin1_novelty3.xlsx')\n",
    "for df in condfdr1.keys():\n",
    "    df1 = condfdr1.get(df)\n",
    "    tr1 = df.split('|')[0]\n",
    "    tr2 = df.split('|')[1]\n",
    "    if tr1 == 'AUD':\n",
    "        db1 = aud_db3\n",
    "    elif tr1 == 'CUD':\n",
    "        db1 = cud_db3\n",
    "    else:\n",
    "        db1 = oud_db3\n",
    "    if tr2 == 'AUD':\n",
    "        db2 = aud_db3\n",
    "    elif tr2 == 'CUD':\n",
    "        db2 = cud_db3\n",
    "    else:\n",
    "        db2 = oud_db3\n",
    "    \n",
    "    #Process query and db for loci search\n",
    "    df2 = df1.iloc[:, :-4]\n",
    "    df2['Novel_leadsnp_db1'] = np.where(df2['LEAD_SNP'].isin(db1['lead_SNP']), 'No', 'Yes')\n",
    "    df2['Novel_leadsnp_db2'] = np.where(df2['LEAD_SNP'].isin(db2['lead_SNP']), 'No', 'Yes')\n",
    "    df2[['CHR', 'MinBP', 'MaxBP']] = df2[['CHR', 'MinBP', 'MaxBP']].apply(pd.to_numeric, errors='coerce')\n",
    "    db1[['chromosome', 'min_bp', 'max_bp']] = db1[['chromosome', 'min_bp', 'max_bp']].apply(pd.to_numeric, errors='coerce')\n",
    "    db2[['chromosome', 'min_bp', 'max_bp']] = db2[['chromosome', 'min_bp', 'max_bp']].apply(pd.to_numeric, errors='coerce')\n",
    "    db1 = db1[['chromosome', 'min_bp', 'max_bp']]\n",
    "    db2 = db2[['chromosome', 'min_bp', 'max_bp']]\n",
    "    db1.dropna(inplace=True)\n",
    "    db2.dropna(inplace=True)\n",
    "\n",
    "    # Process for db_tr1\n",
    "    if not db1.iloc[:, 1].isna().any() and not db1.iloc[:, 2].isna().any():\n",
    "        c1 = df2.iloc[:, 1].to_numpy()[:, None]  # Column 2 of dt2\n",
    "        c2 = db1.iloc[:, 0].to_numpy()[:, None]  # Column 2 of db_tr1\n",
    "        a = df2.iloc[:, 4].to_numpy()[:, None]  # Column 4 of dt2\n",
    "        b = df2.iloc[:, 5].to_numpy()[:, None] # Column 5 of dt2\n",
    "        d1 = db1.iloc[:, 1].to_numpy()[:, None]  # Column 3 of db_tr1\n",
    "        d2 = db1.iloc[:, 2].to_numpy()[:, None]  # Column 4 of db_tr1\n",
    "\n",
    "        result1 = (\n",
    "            ((c1 == c2.T) & (a >= d1.T) & (a <= d2.T)) |\n",
    "            ((c1 == c2.T) & (b >= d1.T) & (b <= d2.T)) |\n",
    "            ((c1 == c2.T) & (a <= d1.T) & (b >= d2.T)) |\n",
    "            ((c1 == c2.T) & (a >= d1.T) & (b <= d2.T))\n",
    "        )\n",
    "        df2['Novel_loci_in_db1'] = np.where(result1.any(axis=1), 'No', 'Yes')\n",
    "    \n",
    "    if not db2.iloc[:, 1].isna().any() and not db2.iloc[:, 2].isna().any():\n",
    "        c1 = df2.iloc[:, 1].to_numpy()[:, None]  # Column 2 of dt2\n",
    "        c2 = db2.iloc[:, 0].to_numpy()[:, None]  # Column 2 of db_tr1\n",
    "        a = df2.iloc[:, 4].to_numpy()[:, None]  # Column 4 of dt2\n",
    "        b = df2.iloc[:, 5].to_numpy()[:, None] # Column 5 of dt2\n",
    "        d1 = db2.iloc[:, 1].to_numpy()[:, None]  # Column 3 of db_tr1\n",
    "        d2 = db2.iloc[:, 2].to_numpy()[:, None]  # Column 4 of db_tr1\n",
    "\n",
    "        result2 = (\n",
    "            ((c1 == c2.T) & (a >= d1.T) & (a <= d2.T)) |\n",
    "            ((c1 == c2.T) & (b >= d1.T) & (b <= d2.T)) |\n",
    "            ((c1 == c2.T) & (a <= d1.T) & (b >= d2.T)) |\n",
    "            ((c1 == c2.T) & (a >= d1.T) & (b <= d2.T))\n",
    "        )\n",
    "        df2['Novel_loci_in_db2'] = np.where(result2.any(axis=1), 'No', 'Yes')\n",
    "\n",
    "\n",
    "    df2['Novel_loci'] = np.where((df2['Novel_leadsnp_db1'] == 'Yes') & (df2['Novel_leadsnp_db2'] == 'Yes') & \n",
    "                                (df2['Novel_loci_in_db1'] == 'Yes') & (df2['Novel_loci_in_db2'] == 'Yes'),'Yes','No' )\n",
    "    df2.to_excel(w2,sheet_name=f'{tr1}|{tr2}',index=False)\n",
    "    print(tr1,db1.shape,tr2,db2.shape,df2.head())\n",
    "w2.close()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 70,
   "id": "07b71ccb",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "dict_keys(['CUD_OUD', 'CUD_AUD', 'AUD_OUD'])"
      ]
     },
     "execution_count": 70,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "conjfdr1 = pd.read_excel('fdr_out/conjfdr_lead_snp2gene_effects_fin1_novelty.xlsx',sheet_name=None)\n",
    "conjfdr1.keys()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 72,
   "id": "c45ad9d8",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "   locusnum  CHR    LEAD_SNP    LEAD_BP      MinBP      MaxBP       FDR  \\\n",
      "0         1    6  rs72828517   19036035   19023726   19191459  0.006021   \n",
      "1         2   11  rs11214677  113477081  113464248  113481638  0.033113   \n",
      "\n",
      "   is_locus_lead A1 A2  ...   PVAL_in_CUD  PVAL_in_OUD             Gene  \\\n",
      "0           True  C  T  ...  1.860000e-03     0.000014  ENSG00000137393   \n",
      "1           True  C  T  ...  4.155000e-07     0.000006  ENSG00000166682   \n",
      "\n",
      "   Approved symbol                    Approved name Overall_Gene_Score  \\\n",
      "0          RNF144B         ring finger protein 144B           0.000000   \n",
      "1          TMPRSS5  transmembrane serine protease 5           0.313481   \n",
      "\n",
      "  Novel_SNP Novel_in_db1  Novel_in_db2  Novel_loci  \n",
      "0      True          Yes            No       False  \n",
      "1     False          Yes            No       False  \n",
      "\n",
      "[2 rows x 23 columns]\n",
      "   locusnum  CHR     LEAD_SNP    LEAD_BP      MinBP      MaxBP       FDR  \\\n",
      "0         1    1    rs6690101   66414039   66364465   66547212  0.003342   \n",
      "1         2    1    rs1526480   91209986   91189731   91234126  0.007359   \n",
      "2         3    3   rs61557789   49871201   49609477   49890967  0.026278   \n",
      "3         4    4  rs190653003  100415029  100395414  100638613  0.015596   \n",
      "4         5    5    rs1807017   60127724   59842971   60565864  0.042540   \n",
      "\n",
      "   is_locus_lead A1 A2  ...   PVAL_in_CUD   PVAL_in_AUD             Gene  \\\n",
      "0           True  T  C  ...  3.351000e-05  2.060000e-09  ENSG00000184588   \n",
      "1           True  C  T  ...  5.906000e-10  1.600000e-05  ENSG00000143032   \n",
      "2           True  T  C  ...  1.056000e-08  4.392000e-05  ENSG00000004534   \n",
      "3           True  A  G  ...  1.216000e-02  2.347000e-14  ENSG00000138813   \n",
      "4           True  C  T  ...  2.735000e-05  1.619000e-04  ENSG00000164181   \n",
      "\n",
      "   Approved symbol                       Approved name Overall_Gene_Score  \\\n",
      "0            PDE4B                phosphodiesterase 4B           0.066600   \n",
      "1           BARHL2                BarH like homeobox 2           0.059759   \n",
      "2             RBM6         RNA binding motif protein 6           0.266398   \n",
      "3          C4orf17  chromosome 4 open reading frame 17           0.066398   \n",
      "4           ELOVL7         ELOVL fatty acid elongase 7           0.314286   \n",
      "\n",
      "  Novel_SNP Novel_in_db1  Novel_in_db2  Novel_loci  \n",
      "0      True           No            No       False  \n",
      "1     False          Yes           Yes       False  \n",
      "2      True          Yes           Yes        True  \n",
      "3      True          Yes            No       False  \n",
      "4      True          Yes           Yes        True  \n",
      "\n",
      "[5 rows x 23 columns]\n",
      "   locusnum  CHR    LEAD_SNP    LEAD_BP      MinBP      MaxBP       FDR  \\\n",
      "0         1    2  rs13428598  144250487  144145478  144272229  0.010449   \n",
      "1         2    6   rs9350100   19076417   18991559   19128832  0.033773   \n",
      "2         3    6   rs1799971  154360797  154269510  154390607  0.006017   \n",
      "3         4    7   rs3812281  135082751  135050259  135221170  0.007410   \n",
      "4         5    8   rs9297901   93036795   92976563   93180965  0.021191   \n",
      "\n",
      "   is_locus_lead A1 A2  ...   PVAL_in_AUD   PVAL_in_OUD             Gene  \\\n",
      "0           True  T  C  ...  1.596000e-03  1.488800e-05  ENSG00000075884   \n",
      "1           True  C  T  ...  8.980000e-07  4.744910e-05  ENSG00000137393   \n",
      "2           True  G  A  ...  1.502000e-05  7.684630e-13  ENSG00000112038   \n",
      "3           True  C  T  ...  5.174000e-07  2.648640e-05  ENSG00000243317   \n",
      "4           True  C  T  ...  5.532000e-04  1.091090e-04  ENSG00000079102   \n",
      "\n",
      "   Approved symbol                                 Approved name  \\\n",
      "0         ARHGAP15              Rho GTPase activating protein 15   \n",
      "1          RNF144B                      ring finger protein 144B   \n",
      "2            OPRM1                          opioid receptor mu 1   \n",
      "3            STMP1   short transmembrane mitochondrial protein 1   \n",
      "4          RUNX1T1  RUNX1 partner transcriptional co-repressor 1   \n",
      "\n",
      "  Overall_Gene_Score Novel_SNP Novel_in_db1  Novel_in_db2  Novel_loci  \n",
      "0           0.139638      True          Yes            No       False  \n",
      "1           0.013280     False           No            No       False  \n",
      "2           0.205835     False          Yes            No       False  \n",
      "3           0.139437      True          Yes            No       False  \n",
      "4           0.073239      True           No            No       False  \n",
      "\n",
      "[5 rows x 23 columns]\n"
     ]
    }
   ],
   "source": [
    "for df in conjfdr1.keys():\n",
    "    df1 = conjfdr1.get(df)\n",
    "    tr1 = df.split('_')[0]\n",
    "    tr2 = df.split('_')[1]\n",
    "    print(df1.head())"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 75,
   "id": "3aa1b49b",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "CUD (24, 3) OUD (31, 3)    locusnum  CHR    LEAD_SNP    LEAD_BP      MinBP      MaxBP       FDR  \\\n",
      "0         1    6  rs72828517   19036035   19023726   19191459  0.006021   \n",
      "1         2   11  rs11214677  113477081  113464248  113481638  0.033113   \n",
      "\n",
      "   is_locus_lead A1 A2  ...  PVAL_in_OUD             Gene Approved symbol  \\\n",
      "0           True  C  T  ...     0.000014  ENSG00000137393         RNF144B   \n",
      "1           True  C  T  ...     0.000006  ENSG00000166682         TMPRSS5   \n",
      "\n",
      "                     Approved name  Overall_Gene_Score Novel_leadsnp_db1  \\\n",
      "0         ring finger protein 144B            0.000000               Yes   \n",
      "1  transmembrane serine protease 5            0.313481               Yes   \n",
      "\n",
      "  Novel_leadsnp_db2 Novel_loci_in_db1  Novel_loci_in_db2 Novel_loci  \n",
      "0               Yes               Yes                Yes        Yes  \n",
      "1                No               Yes                 No         No  \n",
      "\n",
      "[2 rows x 24 columns]\n",
      "CUD (24, 3) AUD (4, 3)    locusnum  CHR     LEAD_SNP    LEAD_BP      MinBP      MaxBP       FDR  \\\n",
      "0         1    1    rs6690101   66414039   66364465   66547212  0.003342   \n",
      "1         2    1    rs1526480   91209986   91189731   91234126  0.007359   \n",
      "2         3    3   rs61557789   49871201   49609477   49890967  0.026278   \n",
      "3         4    4  rs190653003  100415029  100395414  100638613  0.015596   \n",
      "4         5    5    rs1807017   60127724   59842971   60565864  0.042540   \n",
      "\n",
      "   is_locus_lead A1 A2  ...   PVAL_in_AUD             Gene Approved symbol  \\\n",
      "0           True  T  C  ...  2.060000e-09  ENSG00000184588           PDE4B   \n",
      "1           True  C  T  ...  1.600000e-05  ENSG00000143032          BARHL2   \n",
      "2           True  T  C  ...  4.392000e-05  ENSG00000004534            RBM6   \n",
      "3           True  A  G  ...  2.347000e-14  ENSG00000138813         C4orf17   \n",
      "4           True  C  T  ...  1.619000e-04  ENSG00000164181          ELOVL7   \n",
      "\n",
      "                        Approved name  Overall_Gene_Score Novel_leadsnp_db1  \\\n",
      "0                phosphodiesterase 4B            0.066600               Yes   \n",
      "1                BarH like homeobox 2            0.059759               Yes   \n",
      "2         RNA binding motif protein 6            0.266398               Yes   \n",
      "3  chromosome 4 open reading frame 17            0.066398               Yes   \n",
      "4         ELOVL fatty acid elongase 7            0.314286               Yes   \n",
      "\n",
      "  Novel_leadsnp_db2 Novel_loci_in_db1  Novel_loci_in_db2 Novel_loci  \n",
      "0               Yes                No                 No         No  \n",
      "1               Yes                No                Yes         No  \n",
      "2               Yes               Yes                Yes        Yes  \n",
      "3               Yes               Yes                Yes        Yes  \n",
      "4               Yes                No                Yes         No  \n",
      "\n",
      "[5 rows x 24 columns]\n",
      "AUD (4, 3) OUD (31, 3)    locusnum  CHR    LEAD_SNP    LEAD_BP      MinBP      MaxBP       FDR  \\\n",
      "0         1    2  rs13428598  144250487  144145478  144272229  0.010449   \n",
      "1         2    6   rs9350100   19076417   18991559   19128832  0.033773   \n",
      "2         3    6   rs1799971  154360797  154269510  154390607  0.006017   \n",
      "3         4    7   rs3812281  135082751  135050259  135221170  0.007410   \n",
      "4         5    8   rs9297901   93036795   92976563   93180965  0.021191   \n",
      "\n",
      "   is_locus_lead A1 A2  ...   PVAL_in_OUD             Gene Approved symbol  \\\n",
      "0           True  T  C  ...  1.488800e-05  ENSG00000075884        ARHGAP15   \n",
      "1           True  C  T  ...  4.744910e-05  ENSG00000137393         RNF144B   \n",
      "2           True  G  A  ...  7.684630e-13  ENSG00000112038           OPRM1   \n",
      "3           True  C  T  ...  2.648640e-05  ENSG00000243317           STMP1   \n",
      "4           True  C  T  ...  1.091090e-04  ENSG00000079102         RUNX1T1   \n",
      "\n",
      "                                  Approved name  Overall_Gene_Score  \\\n",
      "0              Rho GTPase activating protein 15            0.139638   \n",
      "1                      ring finger protein 144B            0.013280   \n",
      "2                          opioid receptor mu 1            0.205835   \n",
      "3   short transmembrane mitochondrial protein 1            0.139437   \n",
      "4  RUNX1 partner transcriptional co-repressor 1            0.073239   \n",
      "\n",
      "  Novel_leadsnp_db1 Novel_leadsnp_db2 Novel_loci_in_db1  Novel_loci_in_db2  \\\n",
      "0               Yes               Yes               Yes                 No   \n",
      "1               Yes               Yes               Yes                Yes   \n",
      "2               Yes               Yes               Yes                Yes   \n",
      "3               Yes               Yes               Yes                Yes   \n",
      "4               Yes               Yes               Yes                 No   \n",
      "\n",
      "  Novel_loci  \n",
      "0         No  \n",
      "1        Yes  \n",
      "2        Yes  \n",
      "3        Yes  \n",
      "4         No  \n",
      "\n",
      "[5 rows x 24 columns]\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/var/folders/g7/p8558cx50s5gdqg8zvqrdf1h0000gn/T/ipykernel_1874/77464314.py:28: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  db1.dropna(inplace=True)\n",
      "/var/folders/g7/p8558cx50s5gdqg8zvqrdf1h0000gn/T/ipykernel_1874/77464314.py:29: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  db2.dropna(inplace=True)\n",
      "/var/folders/g7/p8558cx50s5gdqg8zvqrdf1h0000gn/T/ipykernel_1874/77464314.py:28: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  db1.dropna(inplace=True)\n",
      "/var/folders/g7/p8558cx50s5gdqg8zvqrdf1h0000gn/T/ipykernel_1874/77464314.py:29: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  db2.dropna(inplace=True)\n",
      "/var/folders/g7/p8558cx50s5gdqg8zvqrdf1h0000gn/T/ipykernel_1874/77464314.py:28: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  db1.dropna(inplace=True)\n",
      "/var/folders/g7/p8558cx50s5gdqg8zvqrdf1h0000gn/T/ipykernel_1874/77464314.py:29: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  db2.dropna(inplace=True)\n"
     ]
    }
   ],
   "source": [
    "w2 = ExcelWriter('fdr_out/conjfdr_lead_snp2gene_effects_fin1_novelty3.xlsx')\n",
    "for df in conjfdr1.keys():\n",
    "    df1 = conjfdr1.get(df)\n",
    "    tr1 = df.split('_')[0]\n",
    "    tr2 = df.split('_')[1]\n",
    "    if tr1 == 'AUD':\n",
    "        db1 = aud_db3\n",
    "    elif tr1 == 'CUD':\n",
    "        db1 = cud_db3\n",
    "    else:\n",
    "        db1 = oud_db3\n",
    "    if tr2 == 'AUD':\n",
    "        db2 = aud_db3\n",
    "    elif tr2 == 'CUD':\n",
    "        db2 = cud_db3\n",
    "    else:\n",
    "        db2 = oud_db3\n",
    "    df2 = df1.iloc[:, :-4]\n",
    "    df2['Novel_leadsnp_db1'] = np.where(df2['LEAD_SNP'].isin(db1['lead_SNP']), 'No', 'Yes')\n",
    "    df2['Novel_leadsnp_db2'] = np.where(df2['LEAD_SNP'].isin(db2['lead_SNP']), 'No', 'Yes')\n",
    "    \n",
    "    #Process db and query for loci search\n",
    "    df2[['CHR', 'MinBP', 'MaxBP']] = df2[['CHR', 'MinBP', 'MaxBP']].apply(pd.to_numeric, errors='coerce')\n",
    "    db1[['chromosome', 'min_bp', 'max_bp']] = db1[['chromosome', 'min_bp', 'max_bp']].apply(pd.to_numeric, errors='coerce')\n",
    "    db2[['chromosome', 'min_bp', 'max_bp']] = db2[['chromosome', 'min_bp', 'max_bp']].apply(pd.to_numeric, errors='coerce')\n",
    "    db1 = db1[['chromosome', 'min_bp', 'max_bp']]\n",
    "    db2 = db2[['chromosome', 'min_bp', 'max_bp']]\n",
    "    db1.dropna(inplace=True)\n",
    "    db2.dropna(inplace=True)\n",
    "\n",
    "    # Process for db_tr1\n",
    "    if not db1.iloc[:, 1].isna().any() and not db1.iloc[:, 2].isna().any():\n",
    "        c1 = df2.iloc[:, 1].to_numpy()[:, None]  # Column 2 of dt2\n",
    "        c2 = db1.iloc[:, 0].to_numpy()[:, None]  # Column 2 of db_tr1\n",
    "        a = df2.iloc[:, 4].to_numpy()[:, None]  # Column 4 of dt2\n",
    "        b = df2.iloc[:, 5].to_numpy()[:, None] # Column 5 of dt2\n",
    "        d1 = db1.iloc[:, 1].to_numpy()[:, None]  # Column 3 of db_tr1\n",
    "        d2 = db1.iloc[:, 2].to_numpy()[:, None]  # Column 4 of db_tr1\n",
    "\n",
    "        result1 = (\n",
    "            ((c1 == c2.T) & (a >= d1.T) & (a <= d2.T)) |\n",
    "            ((c1 == c2.T) & (b >= d1.T) & (b <= d2.T)) |\n",
    "            ((c1 == c2.T) & (a <= d1.T) & (b >= d2.T)) |\n",
    "            ((c1 == c2.T) & (a >= d1.T) & (b <= d2.T))\n",
    "        )\n",
    "        df2['Novel_loci_in_db1'] = np.where(result1.any(axis=1), 'No', 'Yes')\n",
    "    \n",
    "    if not db2.iloc[:, 1].isna().any() and not db2.iloc[:, 2].isna().any():\n",
    "        c1 = df2.iloc[:, 1].to_numpy()[:, None]  # Column 2 of dt2\n",
    "        c2 = db2.iloc[:, 0].to_numpy()[:, None]  # Column 2 of db_tr1\n",
    "        a = df2.iloc[:, 4].to_numpy()[:, None]  # Column 4 of dt2\n",
    "        b = df2.iloc[:, 5].to_numpy()[:, None] # Column 5 of dt2\n",
    "        d1 = db2.iloc[:, 1].to_numpy()[:, None]  # Column 3 of db_tr1\n",
    "        d2 = db2.iloc[:, 2].to_numpy()[:, None]  # Column 4 of db_tr1\n",
    "\n",
    "        result2 = (\n",
    "            ((c1 == c2.T) & (a >= d1.T) & (a <= d2.T)) |\n",
    "            ((c1 == c2.T) & (b >= d1.T) & (b <= d2.T)) |\n",
    "            ((c1 == c2.T) & (a <= d1.T) & (b >= d2.T)) |\n",
    "            ((c1 == c2.T) & (a >= d1.T) & (b <= d2.T))\n",
    "        )\n",
    "        df2['Novel_loci_in_db2'] = np.where(result2.any(axis=1), 'No', 'Yes')\n",
    "\n",
    "\n",
    "    df2['Novel_loci'] = np.where((df2['Novel_leadsnp_db1'] == 'Yes') & (df2['Novel_leadsnp_db2'] == 'Yes') & \n",
    "                                (df2['Novel_loci_in_db1'] == 'Yes') & (df2['Novel_loci_in_db2'] == 'Yes'),'Yes','No' )\n",
    "    df2.to_excel(w2,sheet_name=f'{tr1}_{tr2}',index=False)\n",
    "    print(tr1,db1.shape,tr2,db2.shape,df2.head())\n",
    "w2.close()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 77,
   "id": "70a3c70f",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>ID</th>\n",
       "      <th>RSID</th>\n",
       "      <th>CHR</th>\n",
       "      <th>BP</th>\n",
       "      <th>A1</th>\n",
       "      <th>A2</th>\n",
       "      <th>Neff</th>\n",
       "      <th>Z</th>\n",
       "      <th>P</th>\n",
       "      <th>Direction</th>\n",
       "      <th>HetISq</th>\n",
       "      <th>HetChiSq</th>\n",
       "      <th>HetDf</th>\n",
       "      <th>HetPVal</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>9:72398726:A:C</td>\n",
       "      <td>rs144293193</td>\n",
       "      <td>9</td>\n",
       "      <td>72398726</td>\n",
       "      <td>A</td>\n",
       "      <td>C</td>\n",
       "      <td>52192.72</td>\n",
       "      <td>-0.130</td>\n",
       "      <td>0.89690</td>\n",
       "      <td>+--</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.959</td>\n",
       "      <td>2</td>\n",
       "      <td>0.3755</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>18:3563497:A:G</td>\n",
       "      <td>rs189478541</td>\n",
       "      <td>18</td>\n",
       "      <td>3563497</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>48358.95</td>\n",
       "      <td>-0.745</td>\n",
       "      <td>0.45650</td>\n",
       "      <td>?-+</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.058</td>\n",
       "      <td>1</td>\n",
       "      <td>0.8095</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>2:217147791:A:C</td>\n",
       "      <td>rs6756055</td>\n",
       "      <td>2</td>\n",
       "      <td>217147791</td>\n",
       "      <td>T</td>\n",
       "      <td>G</td>\n",
       "      <td>50799.60</td>\n",
       "      <td>0.387</td>\n",
       "      <td>0.69860</td>\n",
       "      <td>++-</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.277</td>\n",
       "      <td>2</td>\n",
       "      <td>0.8708</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>5:65691902:A:G</td>\n",
       "      <td>rs549927140</td>\n",
       "      <td>5</td>\n",
       "      <td>65691902</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>51761.70</td>\n",
       "      <td>-1.105</td>\n",
       "      <td>0.26920</td>\n",
       "      <td>--+</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.556</td>\n",
       "      <td>2</td>\n",
       "      <td>0.7572</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>17:38855226:A:C</td>\n",
       "      <td>rs140894156</td>\n",
       "      <td>17</td>\n",
       "      <td>38855226</td>\n",
       "      <td>T</td>\n",
       "      <td>G</td>\n",
       "      <td>48015.20</td>\n",
       "      <td>2.526</td>\n",
       "      <td>0.01155</td>\n",
       "      <td>?+?</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000</td>\n",
       "      <td>0</td>\n",
       "      <td>1.0000</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                ID         RSID  CHR         BP A1 A2      Neff      Z  \\\n",
       "0   9:72398726:A:C  rs144293193    9   72398726  A  C  52192.72 -0.130   \n",
       "1   18:3563497:A:G  rs189478541   18    3563497  T  C  48358.95 -0.745   \n",
       "2  2:217147791:A:C    rs6756055    2  217147791  T  G  50799.60  0.387   \n",
       "3   5:65691902:A:G  rs549927140    5   65691902  T  C  51761.70 -1.105   \n",
       "4  17:38855226:A:C  rs140894156   17   38855226  T  G  48015.20  2.526   \n",
       "\n",
       "         P Direction  HetISq  HetChiSq  HetDf  HetPVal  \n",
       "0  0.89690       +--     0.0     1.959      2   0.3755  \n",
       "1  0.45650       ?-+     0.0     0.058      1   0.8095  \n",
       "2  0.69860       ++-     0.0     0.277      2   0.8708  \n",
       "3  0.26920       --+     0.0     0.556      2   0.7572  \n",
       "4  0.01155       ?+?     0.0     0.000      0   1.0000  "
      ]
     },
     "execution_count": 77,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "aud_smt1 = pd.read_csv('replication/AUD.AFR.PGC_MVP_UKB.1.190622.metal.processed.tsv.gz',sep='\\s+')\n",
    "aud_smt1.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 78,
   "id": "a6a289ed",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(9903494, 14)"
      ]
     },
     "execution_count": 78,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "aud_smt1.shape"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 79,
   "id": "941e2273",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CHR</th>\n",
       "      <th>POS</th>\n",
       "      <th>RSID</th>\n",
       "      <th>EffectAllele</th>\n",
       "      <th>OtherAllele</th>\n",
       "      <th>Z</th>\n",
       "      <th>P</th>\n",
       "      <th>N</th>\n",
       "      <th>CaseN</th>\n",
       "      <th>ControlN</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>10</td>\n",
       "      <td>101759979</td>\n",
       "      <td>rs138880521</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>1.225</td>\n",
       "      <td>0.2204</td>\n",
       "      <td>4442</td>\n",
       "      <td>1505</td>\n",
       "      <td>2937</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>10</td>\n",
       "      <td>101759992</td>\n",
       "      <td>rs11596870</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>-0.878</td>\n",
       "      <td>0.3797</td>\n",
       "      <td>6970</td>\n",
       "      <td>2472</td>\n",
       "      <td>4498</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>10</td>\n",
       "      <td>101760107</td>\n",
       "      <td>rs74152907</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>-0.696</td>\n",
       "      <td>0.4864</td>\n",
       "      <td>6844</td>\n",
       "      <td>2412</td>\n",
       "      <td>4432</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>10</td>\n",
       "      <td>101760269</td>\n",
       "      <td>rs116820614</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>0.408</td>\n",
       "      <td>0.6830</td>\n",
       "      <td>7322</td>\n",
       "      <td>2627</td>\n",
       "      <td>4695</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>10</td>\n",
       "      <td>101760622</td>\n",
       "      <td>rs11190358</td>\n",
       "      <td>A</td>\n",
       "      <td>C</td>\n",
       "      <td>0.483</td>\n",
       "      <td>0.6292</td>\n",
       "      <td>7000</td>\n",
       "      <td>2483</td>\n",
       "      <td>4517</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   CHR        POS         RSID EffectAllele OtherAllele      Z       P     N  \\\n",
       "0   10  101759979  rs138880521            G           A  1.225  0.2204  4442   \n",
       "1   10  101759992   rs11596870            C           T -0.878  0.3797  6970   \n",
       "2   10  101760107   rs74152907            C           T -0.696  0.4864  6844   \n",
       "3   10  101760269  rs116820614            G           A  0.408  0.6830  7322   \n",
       "4   10  101760622   rs11190358            A           C  0.483  0.6292  7000   \n",
       "\n",
       "   CaseN  ControlN  \n",
       "0   1505      2937  \n",
       "1   2472      4498  \n",
       "2   2412      4432  \n",
       "3   2627      4695  \n",
       "4   2483      4517  "
      ]
     },
     "execution_count": 79,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "cud_smt1 = pd.read_csv('replication/PGC_CUD_2020_AFR.sumstats.gz',sep='\\s+')\n",
    "cud_smt1.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 80,
   "id": "aea05ff7",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(12645574, 10)"
      ]
     },
     "execution_count": 80,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "cud_smt1.shape"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 81,
   "id": "9fe1b95d",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>SNP_ID</th>\n",
       "      <th>Chromosome</th>\n",
       "      <th>Position</th>\n",
       "      <th>Allele_1</th>\n",
       "      <th>Allele_2</th>\n",
       "      <th>Effect_allele</th>\n",
       "      <th>EAF</th>\n",
       "      <th>Sample_size</th>\n",
       "      <th>OR</th>\n",
       "      <th>SE</th>\n",
       "      <th>Test_statistic</th>\n",
       "      <th>P_value</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>rs7545818</td>\n",
       "      <td>1</td>\n",
       "      <td>100002882</td>\n",
       "      <td>T</td>\n",
       "      <td>G</td>\n",
       "      <td>T</td>\n",
       "      <td>0.646547</td>\n",
       "      <td>88497</td>\n",
       "      <td>0.994339</td>\n",
       "      <td>0.017071</td>\n",
       "      <td>-0.332567</td>\n",
       "      <td>0.739461</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>rs6698430</td>\n",
       "      <td>1</td>\n",
       "      <td>100009669</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>0.642837</td>\n",
       "      <td>88497</td>\n",
       "      <td>0.993576</td>\n",
       "      <td>0.017037</td>\n",
       "      <td>-0.378256</td>\n",
       "      <td>0.705241</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>rs1416883</td>\n",
       "      <td>1</td>\n",
       "      <td>100010999</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>0.653375</td>\n",
       "      <td>88497</td>\n",
       "      <td>0.994357</td>\n",
       "      <td>0.016789</td>\n",
       "      <td>-0.337040</td>\n",
       "      <td>0.736087</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>rs1416882</td>\n",
       "      <td>1</td>\n",
       "      <td>100011253</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>0.643285</td>\n",
       "      <td>88497</td>\n",
       "      <td>0.993754</td>\n",
       "      <td>0.017026</td>\n",
       "      <td>-0.368009</td>\n",
       "      <td>0.712866</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>rs2392043</td>\n",
       "      <td>1</td>\n",
       "      <td>100015373</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>0.645206</td>\n",
       "      <td>88497</td>\n",
       "      <td>0.994300</td>\n",
       "      <td>0.017101</td>\n",
       "      <td>-0.334269</td>\n",
       "      <td>0.738177</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "      SNP_ID  Chromosome   Position Allele_1 Allele_2 Effect_allele       EAF  \\\n",
       "0  rs7545818           1  100002882        T        G             T  0.646547   \n",
       "1  rs6698430           1  100009669        T        C             T  0.642837   \n",
       "2  rs1416883           1  100010999        G        A             G  0.653375   \n",
       "3  rs1416882           1  100011253        G        A             G  0.643285   \n",
       "4  rs2392043           1  100015373        T        C             T  0.645206   \n",
       "\n",
       "   Sample_size        OR        SE  Test_statistic   P_value  \n",
       "0        88497  0.994339  0.017071       -0.332567  0.739461  \n",
       "1        88497  0.993576  0.017037       -0.378256  0.705241  \n",
       "2        88497  0.994357  0.016789       -0.337040  0.736087  \n",
       "3        88497  0.993754  0.017026       -0.368009  0.712866  \n",
       "4        88497  0.994300  0.017101       -0.334269  0.738177  "
      ]
     },
     "execution_count": 81,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "oud_smt1 = pd.read_csv('replication/OUD_less_stringent.AFR.MVP.DrugAlcDep2021.txt.gz',sep='\\s+')\n",
    "oud_smt1.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 82,
   "id": "4df72db3",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Index(['ID', 'RSID', 'CHR', 'BP', 'A1', 'A2', 'Neff', 'Z', 'P', 'Direction',\n",
       "       'HetISq', 'HetChiSq', 'HetDf', 'HetPVal'],\n",
       "      dtype='object')"
      ]
     },
     "execution_count": 82,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "aud_smt1.columns"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 83,
   "id": "efb897c7",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Index(['CHR', 'POS', 'RSID', 'EffectAllele', 'OtherAllele', 'Z', 'P', 'N',\n",
       "       'CaseN', 'ControlN'],\n",
       "      dtype='object')"
      ]
     },
     "execution_count": 83,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "cud_smt1.columns"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 84,
   "id": "9cd2c4c2",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Index(['SNP_ID', 'Chromosome', 'Position', 'Allele_1', 'Allele_2',\n",
       "       'Effect_allele', 'EAF', 'Sample_size', 'OR', 'SE', 'Test_statistic',\n",
       "       'P_value'],\n",
       "      dtype='object')"
      ]
     },
     "execution_count": 84,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "oud_smt1.columns"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 85,
   "id": "2f880cf6",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Index(['CHR', 'POS', 'RSID', 'A1', 'A2', 'Z', 'P', 'N', 'CaseN', 'ControlN'], dtype='object')"
      ]
     },
     "execution_count": 85,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "cud_smt2 = cud_smt1.rename(columns={'EffectAllele':'A1','OtherAllele':'A2'})\n",
    "cud_smt2.columns"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 86,
   "id": "573028fc",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>SNP_ID</th>\n",
       "      <th>Chromosome</th>\n",
       "      <th>Position</th>\n",
       "      <th>Allele_1</th>\n",
       "      <th>Allele_2</th>\n",
       "      <th>Effect_allele</th>\n",
       "      <th>EAF</th>\n",
       "      <th>Sample_size</th>\n",
       "      <th>OR</th>\n",
       "      <th>SE</th>\n",
       "      <th>Test_statistic</th>\n",
       "      <th>P_value</th>\n",
       "      <th>Z</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>rs7545818</td>\n",
       "      <td>1</td>\n",
       "      <td>100002882</td>\n",
       "      <td>T</td>\n",
       "      <td>G</td>\n",
       "      <td>T</td>\n",
       "      <td>0.646547</td>\n",
       "      <td>88497</td>\n",
       "      <td>0.994339</td>\n",
       "      <td>0.017071</td>\n",
       "      <td>-0.332567</td>\n",
       "      <td>0.739461</td>\n",
       "      <td>-19.481629</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>rs6698430</td>\n",
       "      <td>1</td>\n",
       "      <td>100009669</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>0.642837</td>\n",
       "      <td>88497</td>\n",
       "      <td>0.993576</td>\n",
       "      <td>0.017037</td>\n",
       "      <td>-0.378256</td>\n",
       "      <td>0.705241</td>\n",
       "      <td>-22.202031</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>rs1416883</td>\n",
       "      <td>1</td>\n",
       "      <td>100010999</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>0.653375</td>\n",
       "      <td>88497</td>\n",
       "      <td>0.994357</td>\n",
       "      <td>0.016789</td>\n",
       "      <td>-0.337040</td>\n",
       "      <td>0.736087</td>\n",
       "      <td>-20.074690</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>rs1416882</td>\n",
       "      <td>1</td>\n",
       "      <td>100011253</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>0.643285</td>\n",
       "      <td>88497</td>\n",
       "      <td>0.993754</td>\n",
       "      <td>0.017026</td>\n",
       "      <td>-0.368009</td>\n",
       "      <td>0.712866</td>\n",
       "      <td>-21.613896</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>rs2392043</td>\n",
       "      <td>1</td>\n",
       "      <td>100015373</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>0.645206</td>\n",
       "      <td>88497</td>\n",
       "      <td>0.994300</td>\n",
       "      <td>0.017101</td>\n",
       "      <td>-0.334269</td>\n",
       "      <td>0.738177</td>\n",
       "      <td>-19.546523</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "      SNP_ID  Chromosome   Position Allele_1 Allele_2 Effect_allele       EAF  \\\n",
       "0  rs7545818           1  100002882        T        G             T  0.646547   \n",
       "1  rs6698430           1  100009669        T        C             T  0.642837   \n",
       "2  rs1416883           1  100010999        G        A             G  0.653375   \n",
       "3  rs1416882           1  100011253        G        A             G  0.643285   \n",
       "4  rs2392043           1  100015373        T        C             T  0.645206   \n",
       "\n",
       "   Sample_size        OR        SE  Test_statistic   P_value          Z  \n",
       "0        88497  0.994339  0.017071       -0.332567  0.739461 -19.481629  \n",
       "1        88497  0.993576  0.017037       -0.378256  0.705241 -22.202031  \n",
       "2        88497  0.994357  0.016789       -0.337040  0.736087 -20.074690  \n",
       "3        88497  0.993754  0.017026       -0.368009  0.712866 -21.613896  \n",
       "4        88497  0.994300  0.017101       -0.334269  0.738177 -19.546523  "
      ]
     },
     "execution_count": 86,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "oud_smt1['Z'] = oud_smt1['Test_statistic'] / oud_smt1['SE']\n",
    "oud_smt1.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 87,
   "id": "73791552",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Index(['RSID', 'Chromosome', 'Position', 'Allele_1', 'Allele_2', 'A1', 'EAF',\n",
       "       'Sample_size', 'OR', 'SE', 'Test_statistic', 'P_value', 'Z'],\n",
       "      dtype='object')"
      ]
     },
     "execution_count": 87,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "oud_smt2 = oud_smt1.rename(columns={'SNP_ID':'RSID','Effect_allele':'A1'})\n",
    "oud_smt2.columns"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 98,
   "id": "41046873",
   "metadata": {},
   "outputs": [],
   "source": [
    "from scipy.stats import binom_test"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 126,
   "id": "e5e6b9e5",
   "metadata": {},
   "outputs": [],
   "source": [
    "# Data frame to collect results\n",
    "results_df = pd.DataFrame(columns=[\"Trait1\", \"P_value1\",\"Count_No1\", \"Count_Total1\",\n",
    "                                   \"Trait2\",\"P_value2\", \"Count_No2\",\"Count_Total2\"])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 129,
   "id": "68848383",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/var/folders/g7/p8558cx50s5gdqg8zvqrdf1h0000gn/T/ipykernel_1874/2208053712.py:58: DeprecationWarning: 'binom_test' is deprecated in favour of 'binomtest' from version 1.7.0 and will be removed in Scipy 1.12.0.\n",
      "  p_values[key] = binom_test(x=x1,n=n1, p=0.5, alternative='less')\n",
      "/var/folders/g7/p8558cx50s5gdqg8zvqrdf1h0000gn/T/ipykernel_1874/2208053712.py:68: DeprecationWarning: 'binom_test' is deprecated in favour of 'binomtest' from version 1.7.0 and will be removed in Scipy 1.12.0.\n",
      "  p_values[key2] = binom_test(x=x2,n=n2, p=0.5, alternative='less')\n",
      "/var/folders/g7/p8558cx50s5gdqg8zvqrdf1h0000gn/T/ipykernel_1874/2208053712.py:69: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  results_df = results_df.append({\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "   locusnum  CHR    LEAD_SNP    LEAD_BP      MinBP      MaxBP       FDR  \\\n",
      "0         1    6  rs72828517   19036035   19023726   19191459  0.006021   \n",
      "1         2   11  rs11214677  113477081  113464248  113481638  0.033113   \n",
      "\n",
      "   is_locus_lead A1 A2  ...                    Approved name  \\\n",
      "0           True  C  T  ...         ring finger protein 144B   \n",
      "1           True  C  T  ...  transmembrane serine protease 5   \n",
      "\n",
      "   Overall_Gene_Score A1_in_CUD_AFR  Z_in_CUD_AFR  Z_in_CUD_AFR_corrected  \\\n",
      "0            0.000000             T         0.175                  -0.175   \n",
      "1            0.313481             C         0.285                   0.285   \n",
      "\n",
      "  sign_concordance_CUD_AFR A1_in_OUD_AFR Z_in_OUD_AFR  Z_in_OUD_AFR_corrected  \\\n",
      "0                      Yes             C   -11.091629                  -0.175   \n",
      "1                      Yes             T   107.490634                   0.285   \n",
      "\n",
      "  sign_concordance_OUD_AFR  \n",
      "0                      Yes  \n",
      "1                      Yes  \n",
      "\n",
      "[2 rows x 27 columns] CUD 0.25 OUD 0.25\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/var/folders/g7/p8558cx50s5gdqg8zvqrdf1h0000gn/T/ipykernel_1874/2208053712.py:58: DeprecationWarning: 'binom_test' is deprecated in favour of 'binomtest' from version 1.7.0 and will be removed in Scipy 1.12.0.\n",
      "  p_values[key] = binom_test(x=x1,n=n1, p=0.5, alternative='less')\n",
      "/var/folders/g7/p8558cx50s5gdqg8zvqrdf1h0000gn/T/ipykernel_1874/2208053712.py:68: DeprecationWarning: 'binom_test' is deprecated in favour of 'binomtest' from version 1.7.0 and will be removed in Scipy 1.12.0.\n",
      "  p_values[key2] = binom_test(x=x2,n=n2, p=0.5, alternative='less')\n",
      "/var/folders/g7/p8558cx50s5gdqg8zvqrdf1h0000gn/T/ipykernel_1874/2208053712.py:69: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  results_df = results_df.append({\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "   locusnum  CHR     LEAD_SNP    LEAD_BP      MinBP      MaxBP       FDR  \\\n",
      "0         1    1    rs6690101   66414039   66364465   66547212  0.003342   \n",
      "1         2    1    rs1526480   91209986   91189731   91234126  0.007359   \n",
      "2         3    3   rs61557789   49871201   49609477   49890967  0.026278   \n",
      "3         4    4  rs190653003  100415029  100395414  100638613  0.015596   \n",
      "4         5    5    rs1807017   60127724   59842971   60565864  0.042540   \n",
      "\n",
      "   is_locus_lead A1 A2  ...                       Approved name  \\\n",
      "0           True  T  C  ...                phosphodiesterase 4B   \n",
      "1           True  C  T  ...                BarH like homeobox 2   \n",
      "2           True  T  C  ...         RNA binding motif protein 6   \n",
      "3           True  A  G  ...  chromosome 4 open reading frame 17   \n",
      "4           True  C  T  ...         ELOVL fatty acid elongase 7   \n",
      "\n",
      "   Overall_Gene_Score A1_in_CUD_AFR  Z_in_CUD_AFR  Z_in_CUD_AFR_corrected  \\\n",
      "0            0.066600             T         0.187                   0.187   \n",
      "1            0.059759             T        -0.492                   0.492   \n",
      "2            0.266398             C        -0.414                   0.414   \n",
      "3            0.066398           NaN           NaN                     NaN   \n",
      "4            0.314286             T         0.005                  -0.005   \n",
      "\n",
      "  sign_concordance_CUD_AFR A1_in_AUD_AFR Z_in_AUD_AFR  Z_in_AUD_AFR_corrected  \\\n",
      "0                       No           NaN          NaN                   0.187   \n",
      "1                      Yes             T       -0.853                   0.492   \n",
      "2                      Yes           NaN          NaN                   0.414   \n",
      "3                      NaN             A        0.612                     NaN   \n",
      "4                      Yes           NaN          NaN                  -0.005   \n",
      "\n",
      "  sign_concordance_AUD_AFR  \n",
      "0                       No  \n",
      "1                      Yes  \n",
      "2                      Yes  \n",
      "3                      NaN  \n",
      "4                      Yes  \n",
      "\n",
      "[5 rows x 27 columns] CUD 0.2265625 AUD 0.2265625\n",
      "   locusnum  CHR    LEAD_SNP    LEAD_BP      MinBP      MaxBP       FDR  \\\n",
      "0         1    2  rs13428598  144250487  144145478  144272229  0.010449   \n",
      "1         2    6   rs9350100   19076417   18991559   19128832  0.033773   \n",
      "2         3    6   rs1799971  154360797  154269510  154390607  0.006017   \n",
      "3         4    7   rs3812281  135082751  135050259  135221170  0.007410   \n",
      "4         5    8   rs9297901   93036795   92976563   93180965  0.021191   \n",
      "\n",
      "   is_locus_lead A1 A2  ...                                 Approved name  \\\n",
      "0           True  T  C  ...              Rho GTPase activating protein 15   \n",
      "1           True  C  T  ...                      ring finger protein 144B   \n",
      "2           True  G  A  ...                          opioid receptor mu 1   \n",
      "3           True  C  T  ...   short transmembrane mitochondrial protein 1   \n",
      "4           True  C  T  ...  RUNX1 partner transcriptional co-repressor 1   \n",
      "\n",
      "   Overall_Gene_Score A1_in_AUD_AFR  Z_in_AUD_AFR  Z_in_AUD_AFR_corrected  \\\n",
      "0            0.139638           NaN           NaN                     NaN   \n",
      "1            0.013280           NaN           NaN                     NaN   \n",
      "2            0.205835             A         0.605                  -0.605   \n",
      "3            0.139437             T        -1.738                   1.738   \n",
      "4            0.073239             T         1.361                  -1.361   \n",
      "\n",
      "  sign_concordance_AUD_AFR A1_in_OUD_AFR Z_in_OUD_AFR  Z_in_OUD_AFR_corrected  \\\n",
      "0                      NaN             T   -59.657463                     NaN   \n",
      "1                      NaN           NaN          NaN                     NaN   \n",
      "2                      Yes             G    -0.248645                  -0.605   \n",
      "3                      Yes             C   116.707240                   1.738   \n",
      "4                      Yes             C   -71.941626                  -1.361   \n",
      "\n",
      "  sign_concordance_OUD_AFR  \n",
      "0                      NaN  \n",
      "1                      NaN  \n",
      "2                      Yes  \n",
      "3                      Yes  \n",
      "4                      Yes  \n",
      "\n",
      "[5 rows x 27 columns] AUD 0.015625 OUD 0.015625\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/var/folders/g7/p8558cx50s5gdqg8zvqrdf1h0000gn/T/ipykernel_1874/2208053712.py:58: DeprecationWarning: 'binom_test' is deprecated in favour of 'binomtest' from version 1.7.0 and will be removed in Scipy 1.12.0.\n",
      "  p_values[key] = binom_test(x=x1,n=n1, p=0.5, alternative='less')\n",
      "/var/folders/g7/p8558cx50s5gdqg8zvqrdf1h0000gn/T/ipykernel_1874/2208053712.py:68: DeprecationWarning: 'binom_test' is deprecated in favour of 'binomtest' from version 1.7.0 and will be removed in Scipy 1.12.0.\n",
      "  p_values[key2] = binom_test(x=x2,n=n2, p=0.5, alternative='less')\n",
      "/var/folders/g7/p8558cx50s5gdqg8zvqrdf1h0000gn/T/ipykernel_1874/2208053712.py:69: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  results_df = results_df.append({\n"
     ]
    }
   ],
   "source": [
    "w2 = ExcelWriter('fdr_out/conjfdr_lead_snp2gene_effects_sign_concordance.xlsx')\n",
    "for df in conjfdr1.keys():\n",
    "    df1 = conjfdr1.get(df)\n",
    "    tr1 = df.split('_')[0]\n",
    "    tr2 = df.split('_')[1]\n",
    "    if tr1 == 'AUD':\n",
    "        smt1 = aud_smt1\n",
    "    elif tr1 == 'CUD':\n",
    "        smt1 = cud_smt2\n",
    "    else:\n",
    "        smt1 = oud_smt2\n",
    "    if tr2 == 'AUD':\n",
    "        smt2 = aud_smt1\n",
    "    elif tr2 == 'CUD':\n",
    "        smt2 = cud_smt2\n",
    "    else:\n",
    "        smt2 = oud_smt2\n",
    "    df2 = df1.iloc[:, :-4]\n",
    "    #print(df2.columns,tr1,smt1.columns,tr2,smt2.columns)\n",
    "    #leadsnp4[f'A1_in_{trait1}'] = leadsnp4['CAND_SNP'].map(sm1.set_index('SNP')['A1'])\n",
    "    smt1 = smt1.drop_duplicates('RSID')\n",
    "    smt2 = smt2.drop_duplicates('RSID')\n",
    "    df2[f'A1_in_{tr1}_AFR'] = df2['LEAD_SNP'].map(smt1.set_index('RSID')['A1'])\n",
    "    df2[f'Z_in_{tr1}_AFR'] = df2['LEAD_SNP'].map(smt1.set_index('RSID')['Z'])\n",
    "    df2[f'Z_in_{tr1}_AFR_corrected'] = np.where(df2['A1'] == df2[f'A1_in_{tr1}_AFR'],\n",
    "                                                df2[f'Z_in_{tr1}_AFR'],df2[f'Z_in_{tr1}_AFR'] * -1)\n",
    "    mask = df2[f'Z_in_{tr1}_AFR_corrected'].notna()\n",
    "\n",
    "    # Applying conditional logic using np.where only on rows where Z_in_trait_AFR_corrected is not NaN\n",
    "    df2.loc[mask, f'sign_concordance_{tr1}_AFR'] = np.where(\n",
    "        ((df2.loc[mask, f'Z_in_{tr1}'] > 0) & (df2.loc[mask, f'Z_in_{tr1}_AFR_corrected'] > 0)) |\n",
    "        ((df2.loc[mask, f'Z_in_{tr1}'] < 0) & (df2.loc[mask, f'Z_in_{tr1}_AFR_corrected'] < 0)),\n",
    "        'Yes','No')\n",
    "    \n",
    "    df2[f'A1_in_{tr2}_AFR'] = df2['LEAD_SNP'].map(smt2.set_index('RSID')['A1'])\n",
    "    df2[f'Z_in_{tr2}_AFR'] = df2['LEAD_SNP'].map(smt2.set_index('RSID')['Z'])\n",
    "    df2[f'Z_in_{tr2}_AFR_corrected'] = np.where(df2['A1'] == df2[f'A1_in_{tr1}_AFR'],\n",
    "                                                df2[f'Z_in_{tr1}_AFR'],df2[f'Z_in_{tr1}_AFR'] * -1)\n",
    "    mask = df2[f'Z_in_{tr2}_AFR_corrected'].notna()\n",
    "\n",
    "    # Applying conditional logic using np.where only on rows where Z_in_trait_AFR_corrected is not NaN\n",
    "    df2.loc[mask, f'sign_concordance_{tr2}_AFR'] = np.where(\n",
    "        ((df2.loc[mask, f'Z_in_{tr2}'] > 0) & (df2.loc[mask, f'Z_in_{tr2}_AFR_corrected'] > 0)) |\n",
    "        ((df2.loc[mask, f'Z_in_{tr2}'] < 0) & (df2.loc[mask, f'Z_in_{tr2}_AFR_corrected'] < 0)),\n",
    "        'Yes','No')\n",
    "    \n",
    "    # Create a dictionary to store p-values\n",
    "    p_values = {}\n",
    "\n",
    "    # Compute the number of \"Yes\" entries as successes\n",
    "    x1 = df2.loc[mask, f'sign_concordance_{tr1}_AFR'].eq('No').sum()\n",
    "\n",
    "    # Compute the total number of trials\n",
    "    n1 = df2.loc[mask, f'sign_concordance_{tr1}_AFR'].count()\n",
    "\n",
    "    # Compute the p-value and store it in the dictionary using dynamic key\n",
    "    key = f'p_value_{tr1}'\n",
    "    p_values[key] = binom_test(x=x1,n=n1, p=0.5, alternative='less')\n",
    "    \n",
    "    # Compute the number of \"Yes\" entries as successes\n",
    "    x2 = df2.loc[mask, f'sign_concordance_{tr2}_AFR'].eq('No').sum()\n",
    "\n",
    "    # Compute the total number of trials\n",
    "    n2 = df2.loc[mask, f'sign_concordance_{tr2}_AFR'].count()\n",
    "\n",
    "    # Compute the p-value and store it in the dictionary using dynamic key\n",
    "    key2 = f'p_value_{tr2}'\n",
    "    p_values[key2] = binom_test(x=x2,n=n2, p=0.5, alternative='less')\n",
    "    results_df = results_df.append({\n",
    "        \"Trait1\": tr1, \n",
    "        \"P_value1\": p_values[key],\n",
    "        \"Count_No1\": x1,\n",
    "        \"Count_Total1\": n1,\n",
    "        \"Trait2\": tr2, \n",
    "        \"P_value2\": p_values[key2],\n",
    "        \"Count_No2\": x2,\n",
    "        \"Count_Total2\": n2 \n",
    "    }, ignore_index=True)\n",
    "    df2.to_excel(w2,sheet_name=f'{tr1}_{tr2}',index=False)\n",
    "    results_df.to_excel(w2, sheet_name=\"binomial_summary\", index=False)\n",
    "\n",
    "\n",
    "    print(df2.head(),tr1,p_values[key],tr2,p_values[key2])\n",
    "w2.close()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 130,
   "id": "2d0c8d37",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/var/folders/g7/p8558cx50s5gdqg8zvqrdf1h0000gn/T/ipykernel_1874/2852803761.py:58: DeprecationWarning: 'binom_test' is deprecated in favour of 'binomtest' from version 1.7.0 and will be removed in Scipy 1.12.0.\n",
      "  p_values[key] = binom_test(x=x1,n=n1, p=0.5, alternative='less')\n",
      "/var/folders/g7/p8558cx50s5gdqg8zvqrdf1h0000gn/T/ipykernel_1874/2852803761.py:68: DeprecationWarning: 'binom_test' is deprecated in favour of 'binomtest' from version 1.7.0 and will be removed in Scipy 1.12.0.\n",
      "  p_values[key2] = binom_test(x=x2,n=n2, p=0.5, alternative='less')\n",
      "/var/folders/g7/p8558cx50s5gdqg8zvqrdf1h0000gn/T/ipykernel_1874/2852803761.py:69: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  results_df = results_df.append({\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "   locusnum  CHR    LEAD_SNP    LEAD_BP      MinBP      MaxBP       FDR  \\\n",
      "0         1    2  rs12616949   22594027   22549441   22606275  0.006116   \n",
      "1         2    5  rs12513581   60533207   60484179   60614879  0.008724   \n",
      "2         3    6    rs975303   19028788   19023726   19191459  0.001397   \n",
      "3         4    6   rs3734540   26463321   26393021   26581168  0.001896   \n",
      "4         5    7   rs1989903  114137940  114015707  114287116  0.001041   \n",
      "\n",
      "   is_locus_lead A1 A2  ...                                     Approved name  \\\n",
      "0           True  A  G  ...                                               NaN   \n",
      "1           True  C  T  ...                       ELOVL fatty acid elongase 7   \n",
      "2           True  G  A  ...                                               NaN   \n",
      "3           True  C  A  ...  high mobility group nucleosomal binding domain 4   \n",
      "4           True  A  G  ...                                   forkhead box P2   \n",
      "\n",
      "   Overall_Gene_Score A1_in_CUD_AFR  Z_in_CUD_AFR  Z_in_CUD_AFR_corrected  \\\n",
      "0                 NaN             G         0.118                  -0.118   \n",
      "1            0.214085             T         0.173                  -0.173   \n",
      "2                 NaN             A         3.015                  -3.015   \n",
      "3            0.426962             A         0.137                  -0.137   \n",
      "4            0.033400             A        -1.323                  -1.323   \n",
      "\n",
      "  sign_concordance_CUD_AFR A1_in_OUD_AFR Z_in_OUD_AFR  Z_in_OUD_AFR_corrected  \\\n",
      "0                       No             A    99.469421                  -0.118   \n",
      "1                      Yes             C  -105.767594                  -0.173   \n",
      "2                      Yes           NaN          NaN                  -3.015   \n",
      "3                       No             C    10.400732                  -0.137   \n",
      "4                      Yes             A   -45.876858                  -1.323   \n",
      "\n",
      "  sign_concordance_OUD_AFR  \n",
      "0                       No  \n",
      "1                      Yes  \n",
      "2                      Yes  \n",
      "3                       No  \n",
      "4                      Yes  \n",
      "\n",
      "[5 rows x 27 columns] CUD 0.36328125 OUD 0.63671875\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/var/folders/g7/p8558cx50s5gdqg8zvqrdf1h0000gn/T/ipykernel_1874/2852803761.py:58: DeprecationWarning: 'binom_test' is deprecated in favour of 'binomtest' from version 1.7.0 and will be removed in Scipy 1.12.0.\n",
      "  p_values[key] = binom_test(x=x1,n=n1, p=0.5, alternative='less')\n",
      "/var/folders/g7/p8558cx50s5gdqg8zvqrdf1h0000gn/T/ipykernel_1874/2852803761.py:68: DeprecationWarning: 'binom_test' is deprecated in favour of 'binomtest' from version 1.7.0 and will be removed in Scipy 1.12.0.\n",
      "  p_values[key2] = binom_test(x=x2,n=n2, p=0.5, alternative='less')\n",
      "/var/folders/g7/p8558cx50s5gdqg8zvqrdf1h0000gn/T/ipykernel_1874/2852803761.py:69: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  results_df = results_df.append({\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "   locusnum  CHR    LEAD_SNP   LEAD_BP     MinBP     MaxBP       FDR  \\\n",
      "0         1    1   rs1392816  66481188  66334537  66485442  0.000036   \n",
      "1         2    1  rs12122743  91220791  91200757  91234126  0.006585   \n",
      "2         3    5    rs173946  60488297  60135962  60550041  0.001336   \n",
      "3         4    5   rs6452787  87712831  87514778  87822672  0.003577   \n",
      "4         5    6  rs72828558  19128832  19003340  19191459  0.003909   \n",
      "\n",
      "   is_locus_lead A1 A2  ...                           Approved name  \\\n",
      "0           True  T  C  ...  leptin receptor overlapping transcript   \n",
      "1           True  G  A  ...                    BarH like homeobox 2   \n",
      "2           True  C  T  ...             ELOVL fatty acid elongase 7   \n",
      "3           True  G  A  ...              transmembrane protein 161B   \n",
      "4           True  C  T  ...                                     NaN   \n",
      "\n",
      "   Overall_Gene_Score A1_in_CUD_AFR  Z_in_CUD_AFR  Z_in_CUD_AFR_corrected  \\\n",
      "0            0.079678             C         0.579                  -0.579   \n",
      "1            0.059759             A        -0.292                   0.292   \n",
      "2            0.240845             T         0.587                  -0.587   \n",
      "3            0.046479             A        -1.217                   1.217   \n",
      "4                 NaN             T         1.298                  -1.298   \n",
      "\n",
      "  sign_concordance_CUD_AFR A1_in_AUD_AFR Z_in_AUD_AFR  Z_in_AUD_AFR_corrected  \\\n",
      "0                      Yes           NaN          NaN                  -0.579   \n",
      "1                      Yes           NaN          NaN                   0.292   \n",
      "2                      Yes           NaN          NaN                  -0.587   \n",
      "3                       No           NaN          NaN                   1.217   \n",
      "4                      Yes             T        0.279                  -1.298   \n",
      "\n",
      "  sign_concordance_AUD_AFR  \n",
      "0                      Yes  \n",
      "1                      Yes  \n",
      "2                      Yes  \n",
      "3                      Yes  \n",
      "4                      Yes  \n",
      "\n",
      "[5 rows x 27 columns] CUD 0.1334228515625 AUD 0.1334228515625\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/var/folders/g7/p8558cx50s5gdqg8zvqrdf1h0000gn/T/ipykernel_1874/2852803761.py:58: DeprecationWarning: 'binom_test' is deprecated in favour of 'binomtest' from version 1.7.0 and will be removed in Scipy 1.12.0.\n",
      "  p_values[key] = binom_test(x=x1,n=n1, p=0.5, alternative='less')\n",
      "/var/folders/g7/p8558cx50s5gdqg8zvqrdf1h0000gn/T/ipykernel_1874/2852803761.py:68: DeprecationWarning: 'binom_test' is deprecated in favour of 'binomtest' from version 1.7.0 and will be removed in Scipy 1.12.0.\n",
      "  p_values[key2] = binom_test(x=x2,n=n2, p=0.5, alternative='less')\n",
      "/var/folders/g7/p8558cx50s5gdqg8zvqrdf1h0000gn/T/ipykernel_1874/2852803761.py:69: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  results_df = results_df.append({\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "   locusnum  CHR     LEAD_SNP    LEAD_BP      MinBP      MaxBP       FDR  \\\n",
      "0         1    3  rs199698239  107981430  107880069  108031094  0.004063   \n",
      "1         2    6   rs72828517   19036035   19023726   19191459  0.006021   \n",
      "2         3    6   rs35984974   27410422   26175866   28397676  0.002924   \n",
      "3         4    6    rs1799971  154360797  154194081  154390607  0.000012   \n",
      "4         5    9  rs185062166  127961612  127780246  127987802  0.004438   \n",
      "\n",
      "   is_locus_lead A1 A2  ...                            Approved name  \\\n",
      "0           True  A  G  ...                HHLA2 member of B7 family   \n",
      "1           True  C  T  ...                 ring finger protein 144B   \n",
      "2           True  A  G  ...                       serine protease 16   \n",
      "3           True  G  A  ...                     opioid receptor mu 1   \n",
      "4           True  A  G  ...  Rab9 effector protein with kelch motifs   \n",
      "\n",
      "   Overall_Gene_Score A1_in_OUD_AFR  Z_in_OUD_AFR  Z_in_OUD_AFR_corrected  \\\n",
      "0            0.253320             G    155.842480             -155.842480   \n",
      "1            0.000000             C    -11.091629              -11.091629   \n",
      "2            0.132797             A     11.043925               11.043925   \n",
      "3            0.205835             G     -0.248645               -0.248645   \n",
      "4            0.199195             A    -47.512840              -47.512840   \n",
      "\n",
      "  sign_concordance_OUD_AFR A1_in_CUD_AFR Z_in_CUD_AFR  Z_in_CUD_AFR_corrected  \\\n",
      "0                      Yes             A       -0.503             -155.842480   \n",
      "1                      Yes             T        0.175              -11.091629   \n",
      "2                       No             G        0.716               11.043925   \n",
      "3                      Yes             A        0.603               -0.248645   \n",
      "4                      Yes             G       -1.442              -47.512840   \n",
      "\n",
      "  sign_concordance_CUD_AFR  \n",
      "0                      Yes  \n",
      "1                      Yes  \n",
      "2                       No  \n",
      "3                      Yes  \n",
      "4                      Yes  \n",
      "\n",
      "[5 rows x 27 columns] OUD 0.08984375 CUD 0.08984375\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/var/folders/g7/p8558cx50s5gdqg8zvqrdf1h0000gn/T/ipykernel_1874/2852803761.py:58: DeprecationWarning: 'binom_test' is deprecated in favour of 'binomtest' from version 1.7.0 and will be removed in Scipy 1.12.0.\n",
      "  p_values[key] = binom_test(x=x1,n=n1, p=0.5, alternative='less')\n",
      "/var/folders/g7/p8558cx50s5gdqg8zvqrdf1h0000gn/T/ipykernel_1874/2852803761.py:68: DeprecationWarning: 'binom_test' is deprecated in favour of 'binomtest' from version 1.7.0 and will be removed in Scipy 1.12.0.\n",
      "  p_values[key2] = binom_test(x=x2,n=n2, p=0.5, alternative='less')\n",
      "/var/folders/g7/p8558cx50s5gdqg8zvqrdf1h0000gn/T/ipykernel_1874/2852803761.py:69: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  results_df = results_df.append({\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "   locusnum  CHR    LEAD_SNP    LEAD_BP      MinBP      MaxBP           FDR  \\\n",
      "0         1    1  rs12031155   53714139   53658317   53752134  9.997570e-03   \n",
      "1         2    2  rs35942385  144208523  144145478  144263280  2.498582e-03   \n",
      "2         3    6  rs72828517   19036035   19023726   19191459  7.457391e-03   \n",
      "3         4    6   rs3094551   29354799   28674322   29354809  2.603033e-04   \n",
      "4         5    6   rs1799971  154360797  154194081  154390607  2.066166e-08   \n",
      "\n",
      "   is_locus_lead A1 A2  ...                     Approved name  \\\n",
      "0           True  T  C  ...  carnitine palmitoyltransferase 2   \n",
      "1           True  T  G  ...  Rho GTPase activating protein 15   \n",
      "2           True  C  T  ...          ring finger protein 144B   \n",
      "3           True  A  G  ...         ZFP57 zinc finger protein   \n",
      "4           True  G  A  ...              opioid receptor mu 1   \n",
      "\n",
      "   Overall_Gene_Score A1_in_OUD_AFR  Z_in_OUD_AFR  Z_in_OUD_AFR_corrected  \\\n",
      "0            0.213481             T    -17.756744              -17.756744   \n",
      "1            0.146278             T    -53.593484              -53.593484   \n",
      "2            0.000000             C    -11.091629              -11.091629   \n",
      "3            0.119517             G     80.524639              -80.524639   \n",
      "4            0.205835             G     -0.248645               -0.248645   \n",
      "\n",
      "  sign_concordance_OUD_AFR A1_in_AUD_AFR Z_in_AUD_AFR  Z_in_AUD_AFR_corrected  \\\n",
      "0                      Yes             T       -1.724              -17.756744   \n",
      "1                      Yes             T       -2.136              -53.593484   \n",
      "2                      Yes           NaN          NaN              -11.091629   \n",
      "3                      Yes             A        1.391              -80.524639   \n",
      "4                      Yes             A        0.605               -0.248645   \n",
      "\n",
      "  sign_concordance_AUD_AFR  \n",
      "0                      Yes  \n",
      "1                      Yes  \n",
      "2                      Yes  \n",
      "3                      Yes  \n",
      "4                      Yes  \n",
      "\n",
      "[5 rows x 27 columns] OUD 0.000244140625 AUD 0.000244140625\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/var/folders/g7/p8558cx50s5gdqg8zvqrdf1h0000gn/T/ipykernel_1874/2852803761.py:58: DeprecationWarning: 'binom_test' is deprecated in favour of 'binomtest' from version 1.7.0 and will be removed in Scipy 1.12.0.\n",
      "  p_values[key] = binom_test(x=x1,n=n1, p=0.5, alternative='less')\n",
      "/var/folders/g7/p8558cx50s5gdqg8zvqrdf1h0000gn/T/ipykernel_1874/2852803761.py:68: DeprecationWarning: 'binom_test' is deprecated in favour of 'binomtest' from version 1.7.0 and will be removed in Scipy 1.12.0.\n",
      "  p_values[key2] = binom_test(x=x2,n=n2, p=0.5, alternative='less')\n",
      "/var/folders/g7/p8558cx50s5gdqg8zvqrdf1h0000gn/T/ipykernel_1874/2852803761.py:69: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  results_df = results_df.append({\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "   locusnum  CHR    LEAD_SNP    LEAD_BP      MinBP      MaxBP       FDR  \\\n",
      "0         1    1   rs2310819   66440096   66334518   66547212  0.000001   \n",
      "1         2    1  rs12145432   73769330   73766431   73991651  0.000226   \n",
      "2         3    1  rs61535100   80827756   80784642   80871734  0.004921   \n",
      "3         4    1   rs2166171   91208451   91189731   91234126  0.000117   \n",
      "4         5    1   rs7544174  236902955  236897248  236905585  0.003473   \n",
      "\n",
      "   is_locus_lead A1 A2  ...                           Approved name  \\\n",
      "0           True  C  T  ...  leptin receptor overlapping transcript   \n",
      "1           True  T  G  ...                                     NaN   \n",
      "2           True  G  A  ...                                     NaN   \n",
      "3           True  C  T  ...                    BarH like homeobox 2   \n",
      "4           True  T  C  ...                         actinin alpha 2   \n",
      "\n",
      "   Overall_Gene_Score A1_in_AUD_AFR  Z_in_AUD_AFR  Z_in_AUD_AFR_corrected  \\\n",
      "0            0.066398             T         0.366                  -0.366   \n",
      "1                 NaN             T         2.398                   2.398   \n",
      "2                 NaN           NaN           NaN                     NaN   \n",
      "3            0.059759           NaN           NaN                     NaN   \n",
      "4            0.166600           NaN           NaN                     NaN   \n",
      "\n",
      "  sign_concordance_AUD_AFR A1_in_CUD_AFR Z_in_CUD_AFR  Z_in_CUD_AFR_corrected  \\\n",
      "0                      Yes             C       -0.362                  -0.366   \n",
      "1                      Yes             G       -1.529                   2.398   \n",
      "2                      NaN             A       -1.673                     NaN   \n",
      "3                      NaN             T       -0.843                     NaN   \n",
      "4                      NaN             T        0.082                     NaN   \n",
      "\n",
      "  sign_concordance_CUD_AFR  \n",
      "0                      Yes  \n",
      "1                      Yes  \n",
      "2                      NaN  \n",
      "3                      NaN  \n",
      "4                      NaN  \n",
      "\n",
      "[5 rows x 27 columns] AUD 0.10766357486136258 CUD 0.025051229866221547\n",
      "   locusnum  CHR    LEAD_SNP    LEAD_BP      MinBP      MaxBP       FDR  \\\n",
      "0         1    4  rs13125440   39422324   39413780   39425248  0.006060   \n",
      "1         2    4   rs2213041  100247351  100246711  100291659  0.000003   \n",
      "2         3    5    rs331757  124844266  124820663  124852240  0.001326   \n",
      "3         4    6   rs7757112  154151020  153995311  154390607  0.003123   \n",
      "4         5    7   rs2551777  135100476  135050259  135221170  0.000006   \n",
      "\n",
      "   is_locus_lead A1 A2  ...  \\\n",
      "0           True  G  A  ...   \n",
      "1           True  A  C  ...   \n",
      "2           True  G  A  ...   \n",
      "3           True  C  T  ...   \n",
      "4           True  C  T  ...   \n",
      "\n",
      "                                       Approved name  Overall_Gene_Score  \\\n",
      "0                        UDP-glucose 6-dehydrogenase            0.172636   \n",
      "1  alcohol dehydrogenase 1C (class I), gamma poly...            0.199799   \n",
      "2                                                NaN                 NaN   \n",
      "3                               opioid receptor mu 1            0.039839   \n",
      "4        short transmembrane mitochondrial protein 1            0.165996   \n",
      "\n",
      "  A1_in_AUD_AFR  Z_in_AUD_AFR  Z_in_AUD_AFR_corrected  \\\n",
      "0           NaN           NaN                     NaN   \n",
      "1             A         3.144                   3.144   \n",
      "2           NaN           NaN                     NaN   \n",
      "3             T        -1.337                   1.337   \n",
      "4             T        -2.226                   2.226   \n",
      "\n",
      "  sign_concordance_AUD_AFR A1_in_OUD_AFR Z_in_OUD_AFR  Z_in_OUD_AFR_corrected  \\\n",
      "0                      NaN           NaN          NaN                     NaN   \n",
      "1                      Yes             A   -30.680590                   3.144   \n",
      "2                      NaN           NaN          NaN                     NaN   \n",
      "3                       No             C   -19.163119                   1.337   \n",
      "4                      Yes           NaN          NaN                   2.226   \n",
      "\n",
      "  sign_concordance_OUD_AFR  \n",
      "0                      NaN  \n",
      "1                      Yes  \n",
      "2                      NaN  \n",
      "3                       No  \n",
      "4                      Yes  \n",
      "\n",
      "[5 rows x 27 columns] AUD 0.109375 OUD 0.109375\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/var/folders/g7/p8558cx50s5gdqg8zvqrdf1h0000gn/T/ipykernel_1874/2852803761.py:58: DeprecationWarning: 'binom_test' is deprecated in favour of 'binomtest' from version 1.7.0 and will be removed in Scipy 1.12.0.\n",
      "  p_values[key] = binom_test(x=x1,n=n1, p=0.5, alternative='less')\n",
      "/var/folders/g7/p8558cx50s5gdqg8zvqrdf1h0000gn/T/ipykernel_1874/2852803761.py:68: DeprecationWarning: 'binom_test' is deprecated in favour of 'binomtest' from version 1.7.0 and will be removed in Scipy 1.12.0.\n",
      "  p_values[key2] = binom_test(x=x2,n=n2, p=0.5, alternative='less')\n",
      "/var/folders/g7/p8558cx50s5gdqg8zvqrdf1h0000gn/T/ipykernel_1874/2852803761.py:69: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  results_df = results_df.append({\n"
     ]
    }
   ],
   "source": [
    "w2 = ExcelWriter('fdr_out/condfdr_lead_snp2gene_effects_sign_concordance.xlsx')\n",
    "for df in condfdr1.keys():\n",
    "    df1 = condfdr1.get(df)\n",
    "    tr1 = df.split('|')[0]\n",
    "    tr2 = df.split('|')[1]\n",
    "    if tr1 == 'AUD':\n",
    "        smt1 = aud_smt1\n",
    "    elif tr1 == 'CUD':\n",
    "        smt1 = cud_smt2\n",
    "    else:\n",
    "        smt1 = oud_smt2\n",
    "    if tr2 == 'AUD':\n",
    "        smt2 = aud_smt1\n",
    "    elif tr2 == 'CUD':\n",
    "        smt2 = cud_smt2\n",
    "    else:\n",
    "        smt2 = oud_smt2\n",
    "    df2 = df1.iloc[:, :-4]\n",
    "    #print(df2.columns,tr1,smt1.columns,tr2,smt2.columns)\n",
    "    #leadsnp4[f'A1_in_{trait1}'] = leadsnp4['CAND_SNP'].map(sm1.set_index('SNP')['A1'])\n",
    "    smt1 = smt1.drop_duplicates('RSID')\n",
    "    smt2 = smt2.drop_duplicates('RSID')\n",
    "    df2[f'A1_in_{tr1}_AFR'] = df2['LEAD_SNP'].map(smt1.set_index('RSID')['A1'])\n",
    "    df2[f'Z_in_{tr1}_AFR'] = df2['LEAD_SNP'].map(smt1.set_index('RSID')['Z'])\n",
    "    df2[f'Z_in_{tr1}_AFR_corrected'] = np.where(df2['A1'] == df2[f'A1_in_{tr1}_AFR'],\n",
    "                                                df2[f'Z_in_{tr1}_AFR'],df2[f'Z_in_{tr1}_AFR'] * -1)\n",
    "    mask = df2[f'Z_in_{tr1}_AFR_corrected'].notna()\n",
    "\n",
    "    # Applying conditional logic using np.where only on rows where Z_in_trait_AFR_corrected is not NaN\n",
    "    df2.loc[mask, f'sign_concordance_{tr1}_AFR'] = np.where(\n",
    "        ((df2.loc[mask, f'Z_in_{tr1}'] > 0) & (df2.loc[mask, f'Z_in_{tr1}_AFR_corrected'] > 0)) |\n",
    "        ((df2.loc[mask, f'Z_in_{tr1}'] < 0) & (df2.loc[mask, f'Z_in_{tr1}_AFR_corrected'] < 0)),\n",
    "        'Yes','No')\n",
    "    \n",
    "    df2[f'A1_in_{tr2}_AFR'] = df2['LEAD_SNP'].map(smt2.set_index('RSID')['A1'])\n",
    "    df2[f'Z_in_{tr2}_AFR'] = df2['LEAD_SNP'].map(smt2.set_index('RSID')['Z'])\n",
    "    df2[f'Z_in_{tr2}_AFR_corrected'] = np.where(df2['A1'] == df2[f'A1_in_{tr1}_AFR'],\n",
    "                                                df2[f'Z_in_{tr1}_AFR'],df2[f'Z_in_{tr1}_AFR'] * -1)\n",
    "    mask = df2[f'Z_in_{tr2}_AFR_corrected'].notna()\n",
    "\n",
    "    # Applying conditional logic using np.where only on rows where Z_in_trait_AFR_corrected is not NaN\n",
    "    df2.loc[mask, f'sign_concordance_{tr2}_AFR'] = np.where(\n",
    "        ((df2.loc[mask, f'Z_in_{tr2}'] > 0) & (df2.loc[mask, f'Z_in_{tr2}_AFR_corrected'] > 0)) |\n",
    "        ((df2.loc[mask, f'Z_in_{tr2}'] < 0) & (df2.loc[mask, f'Z_in_{tr2}_AFR_corrected'] < 0)),\n",
    "        'Yes','No')\n",
    "    \n",
    "    # Create a dictionary to store p-values\n",
    "    p_values = {}\n",
    "\n",
    "    # Compute the number of \"Yes\" entries as successes\n",
    "    x1 = df2.loc[mask, f'sign_concordance_{tr1}_AFR'].eq('No').sum()\n",
    "\n",
    "    # Compute the total number of trials\n",
    "    n1 = df2.loc[mask, f'sign_concordance_{tr1}_AFR'].count()\n",
    "\n",
    "    # Compute the p-value and store it in the dictionary using dynamic key\n",
    "    key = f'p_value_{tr1}'\n",
    "    p_values[key] = binom_test(x=x1,n=n1, p=0.5, alternative='less')\n",
    "    \n",
    "    # Compute the number of \"Yes\" entries as successes\n",
    "    x2 = df2.loc[mask, f'sign_concordance_{tr2}_AFR'].eq('No').sum()\n",
    "\n",
    "    # Compute the total number of trials\n",
    "    n2 = df2.loc[mask, f'sign_concordance_{tr2}_AFR'].count()\n",
    "\n",
    "    # Compute the p-value and store it in the dictionary using dynamic key\n",
    "    key2 = f'p_value_{tr2}'\n",
    "    p_values[key2] = binom_test(x=x2,n=n2, p=0.5, alternative='less')\n",
    "    results_df = results_df.append({\n",
    "        \"Trait1\": tr1, \n",
    "        \"P_value1\": p_values[key],\n",
    "        \"Count_No1\": x1,\n",
    "        \"Count_Total1\": n1,\n",
    "        \"Trait2\": tr2, \n",
    "        \"P_value2\": p_values[key2],\n",
    "        \"Count_No2\": x2,\n",
    "        \"Count_Total2\": n2 \n",
    "    }, ignore_index=True)\n",
    "    df2.to_excel(w2,sheet_name=f'{tr1}|{tr2}',index=False)\n",
    "    results_df.to_excel(w2, sheet_name=\"binomial_summary\", index=False)\n",
    "\n",
    "\n",
    "    print(df2.head(),tr1,p_values[key],tr2,p_values[key2])\n",
    "w2.close()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 121,
   "id": "0bc4d7bf",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/var/folders/g7/p8558cx50s5gdqg8zvqrdf1h0000gn/T/ipykernel_1874/577880601.py:1: DeprecationWarning: 'binom_test' is deprecated in favour of 'binomtest' from version 1.7.0 and will be removed in Scipy 1.12.0.\n",
      "  binom_test(x=0,n=5, p=0.5, alternative='less')\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "0.03125"
      ]
     },
     "execution_count": 121,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "binom_test(x=0,n=5, p=0.5, alternative='less')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "1944ba0c",
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.10.10"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
